# The genetic and biochemical determinants of mRNA degradation rates in mammals

Vikram Agarwal $^{1,\dagger}$ , David Kelley $^{1,\ddagger}$

<sup>1</sup>Calico Life Sciences LLC, South San Francisco, CA 94080, USA

$^{\dagger}$ Correspondence to Vikram Agarwal (yagar@calicolabs.com) and David Kelley (drk@calicolabs.com)

# ABSTRACT

Background: Degradation rate is a fundamental aspect of mRNA metabolism, and the factors governing it remain poorly characterized. Understanding the genetic and biochemical determinants of mRNA half-life would enable a more precise identification of variants that perturb gene expression through post-transcriptional gene regulatory mechanisms.

Results: Here, we establish a compendium of 54 human and 27 mouse transcriptome-wide mRNA decay rate datasets. A meta-analysis of these data identified a prevalence of technical noise and measurement bias, induced partially by the underlying experimental strategy. Correcting for these biases allowed us to derive more precise, consensus measurements of half-life which exhibit enhanced consistency between species. We trained substantially improved statistical models based upon genetic and biochemical features to better predict half-life and characterize the factors molding it. Our state-of-the-art model, Saluki, is a hybrid convolutional and recurrent deep neural network which relies only upon an mRNA sequence annotated with coding frame and splice sites to predict half-life (r=0.77). Saluki predicts the impact of RNA sequences and genetic mutations therein on mRNA stability, in agreement with functional measurements derived from massively parallel reporter assays.

Conclusions: Our work produces a more robust “ground truth” with regards to transcriptome-wide mRNA half-lives in mammalian cells. Using these consolidated measurements, we trained a model that is over  $50\%$  more accurate in predicting half-life from sequence than existing models. Our best model, Saluki, succinctly captures many of the known determinants of mRNA half-life and can be rapidly deployed to predict the functional consequences of arbitrary mutations in the transcriptome.

# KEYWORDS

mRNA stability, mRNA half-life, deep neural networks, post-transcriptional gene regulation

# INTRODUCTION

The steady-state level of RNA is governed by two opposing forces: the rate of transcription and the rate of decay. While much headway has been made in the problem of predicting steady-state mRNA abundances through the lens of DNA-encoded features that influence transcription rates [1-5], relatively less is known about the RNA-encoded determinants that govern decay rates. Models to predict half-life from RNA sequence in mammals (achieving  $r^2 = 0.20$  and  $r^2 = 0.39$ ) [6,7] have severely lagged behind the performance of those in yeast (achieving  $r^2 = 0.59$ ) [8]. Integrating the modeling of transcription and RNA decay into a unified model to predict steady-state mRNA levels [2] from genetic sequences would elucidate the full spectrum of regulatory functions of non-coding DNA. Consequently, this would advance the goals of providing a mechanistic explanation for evolutionary constraint in conserved sequences [9-11], identifying causal eQTLs [12,13], diagnosing pathogenic non-coding genetic variants [14], and designing more stable and effective mRNA therapeutics [15].

The rate of mRNA decay is experimentally measured as its half-life, or the time elapsed until the initial RNA concentration has decreased by half. Experimental measurement of transcriptome-wide half-life can be achieved by one of two strategies: i) the application of transcriptional inhibitors [e.g., Actinomycin D (ActD) and  $\alpha$ -Amanitin] to cells, or ii) a pulse-chase based method of pulsing modified nucleosides (e.g., 4sU, 5EU, and BrU) followed by chasing with unmodified nucleosides to distinguish newly synthesized RNA from pre-existing

RNA [16]. Both strategies are followed by the profiling of RNA levels over a time course, and data for each gene are fit to an exponential decay curve to ascertain the gene's half-life. Although it has long been appreciated that different methods induce measurement bias [16-18], many modern studies that deploy these methodologies fail to acknowledge the prevalence and impact of these biases on result interpretation.

Measurement biases emerge for a multitude of reasons. Transcriptional inhibitors are convenient, but do not necessarily enter all cells to fully block transcription [16]. Moreover, such drugs can lead to cytotoxicity [17] or impact unintended pathways such as translation, thus preventing the attachment of mRNA to ribosomes and resulting in artificially altered mRNA metabolism [16,18]. Pulse-chase methods are inaccurate in the scenario in which the half-life is shorter than the chase period, and cellular parameters such as doubling time and drug uptake further bias half-life measurement [18]. Finally, the incorporation of uridine analogs is thought to be a stochastic process that is proportional to the number of Us in an RNA, leading to RNA-length-dependent labeling and enrichment biases [19,20]. Collectively, the ultimate consequence of these biases is the disagreement between different methods in the physiologically relevant estimate of half-life [16].

It has been postulated that considering an ensemble of different methods would empower a more precise measurement of half-life, providing a path towards circumventing the tradeoffs and limitations among any individual method [18]. While a meta-analysis of half-life datasets in yeast revealed surprisingly discordant results among half-lives measured by different research groups [21], the consistency among half-life datasets in mammalian organisms remains largely

uncharacterized. Collecting such a compendium of datasets would potentially enable the derivation of consensus measurements of cellular mRNA half-life in a fashion that is less obfuscated by technical noise and methodological bias.

A precise measurement of RNA half-life would enable a clear-eyed examination of how different molecular pathways modulate half-life relative to one another. Numerous RNA-binding proteins (RBPs) and sequence-encoded features have been implicated in regulating half-life. Examples include: i) generic features of an mRNA such as its GC content [22], length, and ORF exon junction density [6,7]; ii) the presence of microRNA (miRNA) binding sites [23-25]; iii) codon frequencies and interactions with the translation machinery [8,26-30]; iv) mRNA structure [31,32]; v) Pumilio binding elements [33]; vi) AU-rich elements (AREs) [16,34]; and vii) YTHDF proteins [35] via m6A recognition [36,37].

Attempts to examine the relative contribution of sequence and biochemical features to the specification of half-life [6-8,38-40] have been undermined by half-life measurement biases, and have not exhaustively considered all of the known pathways that affect RNA stability. In this study, we assembled a compendium of 54 human and 27 mouse mRNA half-life datasets to derive more precise, consensus measurements of RNA stability in mammalian cells. Using our enhanced measurements, we derived improved genetic and biochemical models towards the goals of quantifying the relative influence of different pathways and improving the predictability of half-life from such features. Our state-of-the-art model Saluki, a hybrid convolutional and recurrent neural network, is capable of predicting the effects of RNA sequences and genetic

variants therein on RNA stability, in agreement with functional measurements derived from massively parallel reporter assays.

# RESULTS

Comparison of study concordance in the mammalian half-life compendium

To generate a compendium of mammalian half-life datasets, we mined the literature for published human and mouse half-life data. In total, we identified 33 publications reporting transcriptome-wide half-lives, with 16 submitting human data, 14 submitting mouse data, and 3 submitting data from both species (Table 1). Counting individual replicates, this led to a total of 54 human and 27 mouse half-life measurements. The human studies encompassed 10 cell types and 5 measurement procedures, while the mouse studies encompassed 8 cell types and 3 procedures (Table 1). Each of the human studies reported half-lives for between  $\sim 2,500 - 13,000$  genes (Additional file 1: Fig. S1), while those from the mouse reported half-lives for between  $\sim 500 - 18,000$  genes (Additional file 1: Fig. S2a) (Additional file 2: Dataset S1).

In order to evaluate the consistency among studies, we computed the pairwise Spearman correlations of all datasets, using the subset of common genes measured by each pair of studies. Most pairs of human datasets exhibited strong (i.e.,  $\geq 0.6$ ) correlations (Fig. 1). However, the datasets segregated largely into two clusters; the first encompassed a diverse array of studies, cell types, and procedures, and the second encompassed datasets from only five studies [27,41-44] which appeared to be outliers in that they exhibited poor (i.e.,  $\leq 0.5$ ) correlation to most studies other than to those from their own batch (Fig. 1). Most subclusters did not cleanly segregate by cell type or experimental method; rather, they tended to cluster closely by their study or

<table><tr><td>GroupID</td><td>Method</td><td>Species</td><td>Cell types</td><td>Samples</td><td>Reference</td></tr><tr><td>Akimitsu</td><td>BrU (BRIC-seq)</td><td>H</td><td>HeLa</td><td>1</td><td>[45]</td></tr><tr><td>Akimitsu2</td><td>BrU+4sU (Dyrec-seq)</td><td>H</td><td>HeLa</td><td>1</td><td>[46]</td></tr><tr><td>Bazzini</td><td>4sU (SLAM-seq), ActD</td><td>H</td><td>K562, RPE, HeLa, HEK293</td><td>4</td><td>[29]</td></tr><tr><td>Cramer</td><td>4sU (TT-seq)</td><td>H</td><td>K562</td><td>1</td><td>[47]</td></tr><tr><td>Cramer2</td><td>4sU (TT-seq)</td><td>H</td><td>K562</td><td>1</td><td>[43]</td></tr><tr><td>Darnell</td><td>ActD</td><td>H</td><td>HepG2</td><td>1</td><td>[38]</td></tr><tr><td>Dieterich</td><td>4sU</td><td>H</td><td>HEK293, MCF-7</td><td>4</td><td>[48]</td></tr><tr><td>Gejman</td><td>4sU</td><td>H</td><td>LCLs (7 lines)</td><td>15</td><td>[44]</td></tr><tr><td>He</td><td>ActD</td><td>H</td><td>HeLa</td><td>2</td><td>[36]</td></tr><tr><td>Jaffrey</td><td>ActD</td><td>H</td><td>HEK293</td><td>1</td><td>[42]</td></tr><tr><td>Marks</td><td>ActD</td><td>H</td><td>HeLa</td><td>1</td><td>[49]</td></tr><tr><td>Mortazavi</td><td>4sU (Long-TUC-seq)</td><td>H</td><td>GM12878</td><td>1</td><td>[50]</td></tr><tr><td>Oberdoerffer</td><td>BrU (BRIC-seq)</td><td>H</td><td>HeLa</td><td>1</td><td>[51]</td></tr><tr><td>Rinn</td><td>ActD</td><td>H</td><td>K562, H1-ESC</td><td>6</td><td>[52]</td></tr><tr><td>Rissland</td><td>4sU</td><td>H</td><td>HEK293</td><td>4</td><td>[27]</td></tr><tr><td>Shendure</td><td>4sU</td><td>H</td><td>A549</td><td>1</td><td>[53]</td></tr><tr><td>Rissland2</td><td>4sU, ActD, α-Amanitin</td><td>H, M</td><td>HEK293, 3T3</td><td>6, 2</td><td>[41]</td></tr><tr><td>Simon</td><td>4sU (TimeLapse-seq)</td><td>H, M</td><td>K562, MEF</td><td>2, 2</td><td>[54]</td></tr><tr><td>Zimmer</td><td>ActD</td><td>H, M</td><td>B cell, 3T3</td><td>1, 1</td><td>[55]</td></tr><tr><td>Ameres</td><td>4sU (SLAM-seq)</td><td>M</td><td>mESC</td><td>1</td><td>[56]</td></tr><tr><td>Bartel</td><td>ActD</td><td>M</td><td>3T3</td><td>1</td><td>[7]</td></tr><tr><td>Bartel2</td><td>5EU</td><td>M</td><td>3T3</td><td>2</td><td>[57]</td></tr><tr><td>Darnell</td><td>ActD</td><td>M</td><td>mESC</td><td>1</td><td>[58]</td></tr><tr><td>Dolken</td><td>4sU (SLAM-seq)</td><td>M</td><td>M2-10B4 (bone marrow)</td><td>1</td><td>[59]</td></tr><tr><td>Hanna</td><td>ActD</td><td>M</td><td>mESC, mEB</td><td>2</td><td>[60]</td></tr><tr><td>Hu</td><td>ActD</td><td>M</td><td>mESC (E14Tg2a)</td><td>1</td><td>[61]</td></tr><tr><td>Ko</td><td>ActD</td><td>M</td><td>mESC</td><td>4</td><td>[6]</td></tr><tr><td>Koszinowski</td><td>4sU</td><td>M</td><td>3T3</td><td>1</td><td>[62]</td></tr><tr><td>Mattick</td><td>ActD</td><td>M</td><td>Neuro-2a</td><td>1</td><td>[45]</td></tr><tr><td>Regev</td><td>4sU</td><td>M</td><td>Dendritic cells</td><td>1</td><td>[63]</td></tr><tr><td>Regev2</td><td>4sU</td><td>M</td><td>Dendritic cells</td><td>1</td><td>[64]</td></tr><tr><td>Selbach</td><td>4sU</td><td>M</td><td>3T3</td><td>2</td><td>[65]</td></tr><tr><td>Wilusz</td><td>ActD</td><td>M</td><td>C2C12</td><td>3</td><td>[66]</td></tr></table>

Table 1. Overview of studies compiled for meta-analysis, listing each publication by a GroupID, the method of determining half-lives, the species of origin (M: mouse, H: human), the cell type of origin, and number of samples (comma-separated when listing human and mouse, respectively).

![image](https://cdn-mineru.openxlab.org.cn/result/2026-01-06/a0718be8-98e0-4753-8746-6ff7dd4e8c2f/b1305db2d7a670ad79ae35f506977f5b125636497748410f321cf424a8047d20.jpg)



Figure 1. Comparison of half-lives in a compendium of human datasets. Heatmap of the absolute value of the Spearman correlations measured between half-lives derived from each pair of 54 human samples. Absolute values were used to accommodate five samples from four studies [29,38,46,49] whose data were deposited as degradation rates rather than half-lives. Samples are clustered using hierarchical clustering according to the indicated dendrogram. Rows are labeled by the study of origin (Table 1) and colored by the cell type of origin and measurement approach.


laboratory of origin. This indicated the likely presence of batch effects which masked the cell-type specific signal captured by the data. Datasets from the mouse exhibited similar patterns of clustering, except that only two studies [60,64] appeared to be the greatest outliers due to their poor correlation to most other studies (Additional file 1: Fig. S2b).

# Comparison of methodological bias and cell-type specificity captured in the mammalian half-life compendium

Given the wide disparity in reported genes (Additional file 1: Fig. S1 and S2a) and potential existence of outliers in some samples, we pre-processed our gene x sample human and mouse half-life matrices to improve our ability to evaluate sample relatedness and examine possible sources of measurement bias. We standardized the samples in each matrix, used iterative PCA to impute missing gene measurements, and performed quantile-normalization to align the samples into similar distributions. In total, we recovered 13,921 human genes and 14,463 mouse genes in our matrices (Additional file 3: Dataset S3). Finally, we performed PCA to evaluate the relatedness among the 54 human samples. As observed previously (Fig. 1), PC1 identified samples from a single study [44] to be the greatest outliers relative to all other studies (Additional file 1: Fig. S3). We therefore found it parsimonious to assume that measurements from this study were severely biased, and henceforth excluded samples from this study from further analysis.

Reperforming the PCA analysis on the remaining 39 human samples revealed weak to non-existent clustering by cell type, and stronger clustering by measurement method (Fig. 2a). Specifically, PC2 seemed to segregate samples derived from pulse labeling experiments (i.e.,

![image](https://cdn-mineru.openxlab.org.cn/result/2026-01-06/a0718be8-98e0-4753-8746-6ff7dd4e8c2f/04d16622e00001a3ef6e697444d5193b65b3b059f13dcc224edba48f69a24759.jpg)


![image](https://cdn-mineru.openxlab.org.cn/result/2026-01-06/a0718be8-98e0-4753-8746-6ff7dd4e8c2f/36b994d4f6acd747a6cd8ce18679fa2187cd9697895301ae5084dae4d8ccc6a8.jpg)


![image](https://cdn-mineru.openxlab.org.cn/result/2026-01-06/a0718be8-98e0-4753-8746-6ff7dd4e8c2f/79781c104333a5c49ede42d8d5764fa19e66b049866714b06c0459afea92fb99.jpg)


![image](https://cdn-mineru.openxlab.org.cn/result/2026-01-06/a0718be8-98e0-4753-8746-6ff7dd4e8c2f/148703e7dc169a84d1c054471db8a72879a6fde0dd78a7bf6e188463184a2dcc.jpg)


![image](https://cdn-mineru.openxlab.org.cn/result/2026-01-06/a0718be8-98e0-4753-8746-6ff7dd4e8c2f/1781b264362df10f251c0b46ed09a97307e890d351b758871280d5e30d22dadd.jpg)


![image](https://cdn-mineru.openxlab.org.cn/result/2026-01-06/a0718be8-98e0-4753-8746-6ff7dd4e8c2f/61f6b0647a226c8460147a01569d5c1a6383cb8901b6de35cbcd681708769670.jpg)



Figure 2. Assessment of measurement bias and cell-type specificity present in half-life data.



a) PCA of all human samples except those from an outlier study [44], with sample names colored according to cell type and corresponding data point colored according to measurement approach. Axes are labeled according to the percentage of variance among samples explained by the first two PCs. See also Additional file 1: Fig. S3 for the same analysis using all samples. b) Boxplot of sample distributions along PC2, partitioned according to the measurement method (i.e., pulse labeling or transcriptional shutoff). Replicates for the same study were first averaged according to their PC2 value prior to assessing differences between the methods, with statistical differences between distributions evaluated using a two-sided Wilcoxon rank-sum test. c) Evaluation of the Pearson correlations between pairs of half-life samples. Considered in this plot were the subset of pairs of two different studies that interrogated half-lives in either the same cell type or different cell types. Statistical differences between the distributions were evaluated using a one-sided Wilcoxon rank-sum test to assess whether correlations from the same cell type exceeded those from a different cell type. d-e) These panels are the same as those in (a) and (c), respectively, except compare mouse samples. f) Comparison of consensus, cell-type agnostic (i.e., methodology and cell-type independent) measurements of human and mouse half-lives among one-to-one orthologous genes. Half-lives for each species were computed as PC1 of the respective gene x sample matrix. Also indicated are the Pearson (r) and Spearman (rho) correlation values as well as sample size (n) of genes considered. Shown in all boxplots is the median value (bar), 25th and 75th percentiles (box), and 1.5 times the interquartile range (whiskers).


those using 4sU, BrU, and BrU4sU) with those derived from transcriptional shutoff experiments (i.e., those using ActD and  $\alpha$ -Amanitin). After averaging the PC2 values among replicates, we observed a statistically significant difference between the PC2 distributions for these two methodological classes (Fig. 2b), revealing a fundamental inconsistency in these techniques.

To ascertain whether the human data captured cell-type-specific half-lives, we evaluated all pairwise Pearson correlations between half-lives from samples derived from different studies (in order to minimize the influence of inflated correlations among replicates of the same study), but assessed on either the same cell type or different cell types. We hypothesized that we should see stronger correlations among samples evaluating the same cell type; however, we did not observe statistical support for this hypothesis (Fig. 2c), indicating that there was no detectable cell-type-specific signal for half-lives in human cells. Nearly identical results were achieved using Spearman correlations instead of Pearson correlations (data not shown). Given the known tissue-specific roles of miRNAs [67] and other post-transcriptional RBP regulators, we find it more plausible that technical noise and methodological bias obscure the cell-type specificity of half-life measurements relative to the possibility that none exists biologically.

Next, we again used PCA to evaluate the relatedness among the 27 mouse samples. In the mouse, there appeared to be stronger clustering by cell type and measurement methodology along both PC1 and PC2 (Fig. 2d). Indeed, Pearson correlations between pairs of half-life measurements from the same cell type and different study exceeded those from different cell types and different studies (Fig. 2e), suggesting the detection of a cell-type specific signature. Although this observation is encouraging, the limited sample size, coupled with the confounded

nature of measurement methodology and cell type, make it ultimately impossible to decompose the relative influence of methodological bias and cell-type specificity in the measurement of mouse half-lives. In the future, a controlled study measuring half-lives in a number of different cell types with multiple experimental approaches would help to more accurately assess this.

Given our lack of success in identifying clear cell-type-specific measurements of half-life in the human and mouse samples, we instead focused on deriving cell-type-agnostic (i.e., universal) measures of half-life which integrate information across all methodologies and cell types. Towards this end, we computed the first PC in each of our gene x sample matrices to derive consensus measurements of half-life for each gene from each species. The PC1s explained  $62.4\%$  and  $62.6\%$  of variance among genes in each species, respectively, while PC2s explained merely  $6.6\%$  and  $9.8\%$  of variance, so were henceforth not used. Given that this procedure was performed independently in each species, we would expect that any artifacts induced by the procedure would corrupt the evolutionary relationship of measurements between species. However, we instead observed a strong concordance between our consensus half-life measurements among one-to-one orthologs between the two species (Fig. 2f). Our interspecies Pearson correlation of 0.78 greatly exceeds that of 0.61 reported previously in the literature [6]. This indicates that mRNA half-life has been more strongly conserved than previously appreciated between mammalian species separated by  $\sim 75$  million years of evolutionary time. Moreover, this finding also demonstrates our ability to recover a highly precise measurement of half-life that successfully ameliorates the impact of technical noise and methodological bias.

# A genetic model of mRNA half-life

Given our consensus measurements of mRNA half-life, we sought to determine whether it was possible to improve the predictability of half-life from mRNA sequence. Towards this end, we engineered groups of features (Table 2) with the goal of deriving a machine learning (ML)-based regression model to automatically select the subset of pertinent features. Our sequence-derived feature groups consisted of: i) basic mRNA features such as the length and G/C content of different functional regions and ORF exon junction density [2,6,7]; ii)  $k$ -mer frequencies of length 1-7 in the  $5'$  UTR, ORF, or  $3'$  UTR; iii) codon frequencies; iv) predicted repression scores of mammalian-conserved miRNA families [24]; and v) predicted binding of various RBPs to the mRNA sequence by SeqWeaver [68] and DeepRiPE [69]. RBP binding was predicted separately for the  $5'$  UTR, ORF, and  $3'$  UTR because it has previously been observed from cross-linking and immunoprecipitation sequencing (CLIP-seq) that RBPs have a tendency to bind in specific functional regions [70].

Next, we sought to determine which of these sequence-derived groups of features would be useful for predicting half-life. Because of the hierarchical association of each feature to a group (Table 2), we thus evaluated a series of nested models which iteratively considered additional groups of features. Each feature set was fed into a lasso regression model, and the relative model performance was compared between a simpler and more complex group on different folds of held-out data using a 10-fold cross-validation (CV) strategy. This helped to establish whether the inclusion of additional groups was justified. Evaluating the model series on our human half-life data, we observed that: i)  $5^{\prime}$ -UTR  $k$ -mers did not improve the model, ii) ORF  $k$ -mers did not improve the model beyond the simpler consideration of codons, iii) both  $3^{\prime}$ -UTR  $k$ -mers and

<table><tr><td>Feature description</td><td>Feature type</td><td>Data source</td><td># features</td></tr><tr><td>Basic mRNA features such as length and G/C content of 5&#x27; UTR, ORF, and 3&#x27; UTR; intron length; ORF exon junction density</td><td>Sequence-derived</td><td>Custom Perl scripts</td><td>8</td></tr><tr><td>Codon frequencies</td><td>Sequence-derived</td><td>Custom R scripts</td><td>64</td></tr><tr><td>1- to 7-mer frequencies in the 5&#x27; UTR</td><td>Sequence-derived</td><td>Custom R scripts</td><td>21,844</td></tr><tr><td>1- to 7-mer frequencies in the ORF</td><td>Sequence-derived</td><td>Custom R scripts</td><td>21,844</td></tr><tr><td>1- to 7-mer frequencies in the 3&#x27; UTR</td><td>Sequence-derived</td><td>Custom R scripts</td><td>21,844</td></tr><tr><td>Predicted average binding score of human RBPs (in each of 5&#x27; UTR, ORF, and 3&#x27; UTR)</td><td>Sequence-derived</td><td>DeepRiPe predictions [69]</td><td>177</td></tr><tr><td>Predicted average binding score of human and mouse RBPs (in each of 5&#x27; UTR, ORF, and 3&#x27; UTR)</td><td>Sequence-derived</td><td>SeqWeaver predictions [68]</td><td>780</td></tr><tr><td>Number of CLIP peaks for various RBPs</td><td>Biochemical</td><td>ENCORI database [71]</td><td>133</td></tr><tr><td>Number of eCLIP peaks for various RBPs (K562, HepG2, and adrenal gland)</td><td>Biochemical</td><td>ENCORE database [72]</td><td>225</td></tr><tr><td>Number of PAR-CLIP peaks</td><td>Biochemical</td><td>[70]</td><td>146</td></tr><tr><td>Number of CLIP peaks for m6A pathway components (a6A, m6Am, YTHDF2, METTL3, METTL14, and WTAP)</td><td>Biochemical</td><td>[37,42,73–77]</td><td>13</td></tr><tr><td>RIP-seq of diverse RBPs and translational efficiency</td><td>Biochemical</td><td>[36,78–80]</td><td>34</td></tr></table>

Table 2. Summary of features considered in models trained to predict mRNA half-life, with a description of the features considered, feature type (i.e., sequence or biochemical), data source, and number of features in the category. Sequence features were calculated identically for both human and mouse. Biochemical features were calculated only with respect to human data.

predicted miRNA repression scores improved the model, and iv) SeqWeaver predictions of RBP binding improved the model more than those from DeepRiPE (Fig. 3a). Our final model that optimally balanced the tradeoff between complexity and performance was the "BC3MS" model, which considered basic mRNA features, codon frequencies,  $3^{\prime}$ -UTR  $k$ -mers, miRNA repression scores, and SeqWeaver predictions.


a


![image](https://cdn-mineru.openxlab.org.cn/result/2026-01-06/a0718be8-98e0-4753-8746-6ff7dd4e8c2f/72345b8258b1388812061e00f673314b21e93f83ffe3b638e5f8a50ec550a4a1.jpg)



b


![image](https://cdn-mineru.openxlab.org.cn/result/2026-01-06/a0718be8-98e0-4753-8746-6ff7dd4e8c2f/9bad48962c7b9ab5f7b00d6f8148b0ec88ad9d47c3299dd167ba721f931bd26b.jpg)


B: Basic mRNA features (8)

5: 5' UTR 1-7mers (21,844)

O: ORF 1-7mers (21,844)

3:  $3^{\prime}$  UTR 1-7mers (21,844)

C:Codons (62)

M: miRNA target repression (319)

S:SeqWeaver RBP binding (780)

D: DeepRiPE RBP binding (177)


C


![image](https://cdn-mineru.openxlab.org.cn/result/2026-01-06/a0718be8-98e0-4753-8746-6ff7dd4e8c2f/3420518bb4a259deb709ac2f4e0b76831df9c431c42d8d3a19eb3d7130062c85.jpg)



d


![image](https://cdn-mineru.openxlab.org.cn/result/2026-01-06/a0718be8-98e0-4753-8746-6ff7dd4e8c2f/fd4f6398e9ee97772e383bab21602e6571e5344e31492b3ed2f09844e6b2948b.jpg)



Figure 3. Prediction of human half-lives using sequence-encoded features.


a) Performance of trained lasso regression models on each of 10 held-out folds of data. Compared is the relative performance between pairs of nested models which iteratively consider greater numbers of features. Each model is described by a code indicating the features considered. A description of the code is provided in the key, along with the corresponding number of features considered listed in parentheses. An improvement in a more complex model relative to a simpler model was evaluated with a one-sided, paired t-test, adjusted with a Bonferroni correction to account for the total number of hypothesis tests. Features which were ultimately determined to contribute to performance improvement are colored, or are left black if they did not improve the model. b) Shown are the final predictions for the optimal model (i.e., BC3MS) after concatenating the observations for all 10 folds of held-out data. Also indicated are the Pearson (r) and Spearman (rho) correlation values. c) The top 30 ranked model coefficients corresponding to the BC3MS model, trained on the full dataset. Features are colored according to the same key as that in panel (a). d) Pearson correlation matrix between the union of all top 30 features from (c), shown as rows,

and other features sharing a Pearson correlation either  $\leq -0.8$  or  $\geq 0.8$ , shown as columns. Feature names are colored according to the origin of the feature as shown in the same key as panel (a). Hierarchical clustering was used to group features exhibiting similar correlation patterns.

Concatenating the predictions across all folds of held-out data, we observed a correlation of 0.67 between the BC3MS model predictions and observed human half-lives (Fig. 3b). We retrained this model on the full dataset to assess which features contributed most significantly to half-life prediction. The top-30 ranked features of the model consisted primarily of basic mRNA properties, codon frequencies, and predicted miRNA/RBP binding sites (Fig. 3c). Consistent with previous work [2,6,7], ORF exon junction density was the dominant feature. Moreover, the signs of coefficients associated with codon frequencies are consistent with previously published human Codon Stability Coefficients (CSCs) [26,27,29,30], which were computed based upon an isolated Pearson or Spearman correlation between the codon frequencies and half-lives [28]. CSCs have been previously established as a quantitative metric that captures the association between a codon and mRNA stability [26-30]. Our model contextualized relative codon influence with respect to other sequence features in a multiple linear regression framework, and re-ranked their utility in the prediction task.

Next, despite considering over 20,000 possible  $3^{\prime}$ -UTR  $k$ -mers, the model automatically discovered highly conserved regulatory motifs such as UAUUUUAU, the core AU-rich element (ARE) [16,34]; UGUAAAU and its variant UGUAAU, Pumilio binding elements [33]; and AAUAAA, the cleavage and polyadenylation motif involved in alternative polyadenylation [81,82]. The binding of four RBPs emerged as the most useful features, including that of SAFB2 and RBM22 in the ORF as well as HNRNPUL1 and SRSF3 in the  $3^{\prime}$ -UTR. Finally, consistent

with previous work showing that miRNAs weakly impact half-life [7], the model predicted light repressive roles for two miRNAs, miR-19a and miR-25.

Although we find these factors to be promising candidates, we caution that the interpretation of feature selection and coefficient-based ranking is inherently limited by the substantial degree of multicollinearity among features. To guard against the possibility of lasso regression selecting one feature over another through its spuriously higher correlation to half-life, we examined the full set of features strongly correlated to the top 30 selected features (Fig. 3d). While the majority of features were not strongly correlated to other features, we indeed found that SeqWeaver predicted a similar degree of binding among numerous alternative factors such as MTPAP, XRCC6, AKAP8L, RBM15, BUD13, SLTM, SRSF9, HNRNPU, TAF15, BCCIP, CSTF2T, TIA1, LSM11, MBNL, and SRSF4. We consider this full set of RBPs, in conjunction with our earlier set, as candidate post-transcriptional regulators for future experimental investigation.

We repeated the same analyses independently for the mouse, with the goal of devising a genetic model to predict mouse half-lives. While most findings were highly similar between the two species, key differences include: i) DeepRiPE features significantly boosted performance, leading to the optimality of the "BC3MSD" model (Additional file 1: Fig. S4a); ii) the model achieved a slightly lower Pearson correlation of 0.61 (Additional file 1: Fig. S4b); iii) the feature ranking varied, with the inclusion of FMR1, SRSF4, HNRNPM, RBM15, IGF2BP3, KHSRP, MBNL, miR-27a, and FXR2 in the top 30 coefficients (Additional file 1: Fig. S4c);

and additional factors correlated with the top 30 features, such as HNRNPM, SFPQ, SRSF3, TRA2A, SRSF1, EIF3D, DDX6, XRCC6, and SRSF7 (Additional file 1: Fig. S4d).

Finally, we evaluated the ability of the models trained in each species to generalize to the opposite species. Our interspecies comparisons revealed that models tested in the opposite species performed competitively, albeit slightly worse, than models trained within the same species, for both the human and mouse (Additional file 1: Fig. S4e). This indicates that the vast majority of the learned half-life-associated features have strong predictive value more generally across the mammalian phylogeny.

# A biochemical model of mRNA half-life

Given the recent trove of biochemical data evaluating RBP binding, we attempted to identify experimentally supported features that are predictive of mRNA half-life. Towards this goal, we assembled an array of several large-scale datasets measuring RBP binding to develop an ML-based regression model analogous to our genetic model. Our biochemical feature groups consisted of: i) the number of CLIP peaks in an mRNA for various RBPs in the ENCORI database [71], ii) the number of eCLIP peaks for various RBPs in the ENCORE database [72], iii) the number of PAR-CLIP peaks for various RBPs [70], iv) the number of CLIP peaks for m6A pathway components [37,42,73–77], v) and RIP-seq of diverse RBPs and translational efficiency [36,78–80] (Table 2). Implementing the same strategy as before to compare nested models with 10-fold CV, we observed that: i) PAR-CLIP data did not improve the model, ii) RIP-seq data did not improve the model after jointly considering eCLIP and m6A CLIP data, and iii) the remaining datasets all improved the model, with the greatest benefit emerging from

considering the ENCORI database (Fig. 4a). “BEeM” was our best model, which considered basic mRNA features, ENCORI CLIP, eCLIP, and m6A CLIP data.


a


![image](https://cdn-mineru.openxlab.org.cn/result/2026-01-06/a0718be8-98e0-4753-8746-6ff7dd4e8c2f/92db734a8a97f50bd713f3bf85e5a4ca328e7a00b46426a39d9b0b57d3baf9cf.jpg)



b


![image](https://cdn-mineru.openxlab.org.cn/result/2026-01-06/a0718be8-98e0-4753-8746-6ff7dd4e8c2f/535c632c38a43fdaae8acb42d57c4c1bf9b83375f97504893dbf343733bce878.jpg)



B: Basic mRNA features (8) P: PAR-CLIP (144) e: eCLIP (224)  
M: m6A CLIP (13) R: RIP-seq (33) E: ENCORI CLIP (133)



C


![image](https://cdn-mineru.openxlab.org.cn/result/2026-01-06/a0718be8-98e0-4753-8746-6ff7dd4e8c2f/4dd393ad9b8fd9b1e1cbec4ae8a19cd21dfda7d162f46b831cd9520f485b2776.jpg)



d


![image](https://cdn-mineru.openxlab.org.cn/result/2026-01-06/a0718be8-98e0-4753-8746-6ff7dd4e8c2f/a56943a64cf6e3bf2a428f1b5d5b499cbf8f22c0ddfe4f832bec3c6d8124f8fa.jpg)



Figure 4. Prediction of human half-lives using biochemical features. This figure is organized in the same fashion as Fig. 3, except it evaluates features derived from biochemical experiments. All CLIP data is computed as the number of peaks on the full-length transcript, while RIP-seq is represented as a continuous measurement of the enrichment of RBP binding relative to a control IP.


We observed a global correlation of 0.77 between the BEeM model's held-out predictions and observed human half-lives (Fig. 4b). The top-30 ranked features of the model (i.e., retrained on the full dataset) consisted primarily of the ORF exon junction density and dozens of RBPs (Fig. 4c). A number of RBPs consistently emerged between our genetic (Fig. 3c,d and Additional file 1: Fig. S4c,d) and biochemical models, including BUD13, BCCIP, HNRNPM, FMR1, PUM1 (i.e., a Pumilio-element binding protein), ELAVL1 (i.e., an ARE binding protein), TAF15, IGF2BP3, FXR2, and RBM15. Novel components that were not previously captured by the genetic model include LIN28A/B, SND1, PRPF8, FUS, RC3H1, U2AF2, RBFOX2, LARP4, YTHDF1 and its ligand m6A, DDX3X, UPF1, YBX3, CNBP, HNRNPA1, and SUB1 (Fig. 4c). Collectively, these factors were strongly correlated to others which serve as candidate regulators of mRNA stability (Fig. 4d).

Finally, we tested the possibility that a joint genetic and biochemical model might outperform either individually. A lasso regression model jointly trained upon BC3MS (i.e., genetic) and BEeM (i.e., biochemical) features modestly outperformed our BEeM model alone (Additional file 1: Fig. S5a), achieving a Pearson correlation of 0.78 (Additional file 1: Fig. S5b). The biochemical features were dominantly used in this model, although several codons were selected, along with the ARE 7-mer, which minimized the effect size of the coefficient attributable to ELAVL1 (Additional file 1: Fig. S5c,d).

# A deep-learning based genetic model of mRNA half-life

Having compared the performance of genetic and biochemical models, we sought to evaluate whether an alternative learning paradigm might be able to automatically decipher

sequence-based rules that are more predictive of mRNA half-life. Towards this goal, we trained a hybrid convolutional and recurrent deep neural network architecture, called Saluki, which has the advantage of potentially uncovering nonlinear (e.g., cooperative) relationships among motifs and spatial principles with respect to motif positioning. The input to our model was a binary encoding of the mRNA sequence (i.e., up to a maximum of 12,288 nt), concatenated with binary variables for each nucleotide indicating the presence or absence of the first reading frame of a codon and splice site (Fig. 5a). This matrix was fed in to a neural network comprised of 64 1D convolutions (width 5), a max pooling layer (width 2), 6 additional blocks of these aforementioned layers, a recurrent layer consisting of a gated recurrent unit (GRU), and a densely connected layer (Methods, Fig. 5a, and Additional file 1: Fig. S6a). Given that mRNA sequences are variable in length, the  $3^{\prime}$  end of each mRNA was padded with Ns after the transcriptional end site. To account for this property of the input, we left-justified the input matrix, and the GRU was oriented to pass information from the right to left such that the sequence was still most recently considered by the network (i.e., as opposed to the padded Ns) when integrating information from the full sequence to make a prediction. The parameters comprising this common "trunk" of the network were jointly trained on human and mouse data in alternating batches of species-specific data, with two "heads" to predict human and mouse half-lives, respectively [4,5].

We performed an ablation analysis in which the model was evaluated after training it with only sequence as input; sequence and reading frame tracks; and sequence, reading frame, and splice site tracks. This analysis revealed that all forms of information were utilized by the network for Saluki to achieve optimal performance (Additional file 1: Fig. S6b). Training our Saluki model

with the identical folds of data as our lasso regression models enabled us to directly compare their respective performances on each of the 10 folds of held-out data. Averaging the predictions using models derived from each of five independent training runs for each fold, Saluki's performance exceeded that of the genetic lasso regression models for each species (i.e., "BC3MS" for human and "BC3MSD" for mouse), and it performed nearly identically with the human biochemical regression model (i.e., "BEeM") (Fig. 5b). The final human and mouse models displayed correlations of 0.77 and 0.73, respectively (Fig. 5c and Additional file 1: Fig. S6c), suggesting that Saluki potentially learned novel principles of post-transcriptional gene regulation which our simpler linear models were unable to capture.


a


![image](https://cdn-mineru.openxlab.org.cn/result/2026-01-06/a0718be8-98e0-4753-8746-6ff7dd4e8c2f/8ffc0eafb1f42c6009639d2c38dca473326b0882ff8ebdd8f208c126b5bcee99.jpg)



b


![image](https://cdn-mineru.openxlab.org.cn/result/2026-01-06/a0718be8-98e0-4753-8746-6ff7dd4e8c2f/287705ca5cad42f923c115da7381d6c4cf6641a2483622c157475698d1c25aa1.jpg)



C


![image](https://cdn-mineru.openxlab.org.cn/result/2026-01-06/a0718be8-98e0-4753-8746-6ff7dd4e8c2f/5861e7c4844f44dd34215eb6d47a9c70503955b80b52915cf86c2860cbc8506f.jpg)



d


![image](https://cdn-mineru.openxlab.org.cn/result/2026-01-06/a0718be8-98e0-4753-8746-6ff7dd4e8c2f/1d15239d54950dbba0b921ff975e252f52653ae3c144adcb6eb1a4ae924c41be.jpg)



e


![image](https://cdn-mineru.openxlab.org.cn/result/2026-01-06/a0718be8-98e0-4753-8746-6ff7dd4e8c2f/347bdd444d75d1df3d5702218bee2ddeb41bcb833d43f0eb7bd1fb62e1090b38.jpg)



f


![image](https://cdn-mineru.openxlab.org.cn/result/2026-01-06/a0718be8-98e0-4753-8746-6ff7dd4e8c2f/65ad3fbf5c8c7a4112896d85e4af6b5c935791fb938ce02f0d54a8b45bca4a31.jpg)



g


![image](https://cdn-mineru.openxlab.org.cn/result/2026-01-06/a0718be8-98e0-4753-8746-6ff7dd4e8c2f/c379c3c6bab0163ad832e3f6eafe05db44a3f3ecde6e3468aa0775792c4c1e9c.jpg)



h


![image](https://cdn-mineru.openxlab.org.cn/result/2026-01-06/a0718be8-98e0-4753-8746-6ff7dd4e8c2f/cd0c9a82c947aabdaa1a3605a100260aec5203cb20e55fc4321c484d02dc359d.jpg)



Figure 5. State-of-the-art prediction of half-lives and genetic variant functional effects using a sequence-based deep learning model.


a) A hybrid convolutional/recurrent neural network architecture to predict half-life from an input of the RNA sequence, an encoding of the first frame of each codon, and  $5^{\prime}$  splice site junction(s). The deep learning model, called Saluki, was jointly trained on mouse and human half-life data to predict species-specific half-lives. b) Performance of the trained Saluki models on each of 10 held-out folds of data, relative to the corresponding performances from our best genetic (i.e., "BC3MS" for human and "BC3MSD" for mouse, respectively) and biochemical (i.e., "BEeM") lasso regression models. An improvement relative to another model was evaluated with a two-sided, paired t-test. c) Shown are the final predictions after concatenating the observations for all 10 folds of held-out data. Also indicated are the Pearson (r) and Spearman (rho) correlation values. d) Metagenome plot of ISM scores across all mRNAs for percentiles along the  $5^{\prime}$  UTR, ORF, and  $3^{\prime}$  UTR. mRNAs were grouped into one of 4 bins according to their predicted half-lives. For the set of mRNAs within each bin, we plotted the average of the absolute value of the mean predicted effect size (i.e., of the three possible alternative mutations). e) ISM results of two  $3^{\prime}$ -UTR segments from TUBGCP3 and PI4K2B. Partial matches to the AU-rich element (ARE, or "UAUUUAU") and Pumilio/FBF (PUF, or "UGUAHAUA") binding element consensus sequences are

boxed. For each motif, single point mutations resulting in particularly severe or opposite phenotypes are shown alongside annotations reflecting the corresponding ARE and PUF consensus gain or loss events. f) Insertional analysis of motifs discovered by TF-MoDISco [83]. Each motif was inserted into one of 50 positional bins along the  $5^{\prime}$  UTR, ORF, and  $3^{\prime}$  UTR of each mRNA. Indicated is the average predicted change in half-life for each bin plotted along a metagene. g) This panel is the same as panel f), except it performs analysis of 61 codons (excluding the 3 stop codons) inserted into the first reading frame along the length of the ORF. Selected codons are colored, with the rest shown in gray. h) Scatter plot showing the relationship between the mean influence of each codon along the length of the ORF, as predicted by Saluki in panel g), and the mean codon stability coefficient over a set of cell types as observed previously [26]. Also indicated are the Pearson (r) and Spearman (rho) correlation values.

In the hopes of revealing such principles, we tested the predictive behavior of Saluki in different contexts. First, we performed an in silico mutagenesis (ISM) for every human mRNA in our dataset, mutating each reference nucleotide of an mRNA into every alternative allele to evaluate the predicted change in half-life [1,4,5,84]. We generated a metagene plot using these scores, evaluating the absolute value of the mean effect size (i.e., of the three possible alternative mutations) in percentiles along the length of each functional region (Fig. 5d). This analysis revealed that the model predicts a relatively modest influence for the  $5^{\prime}$  UTR relative to the ORF and  $3^{\prime}$  UTR. Moreover, the termini of functional regions, such as the  $3^{\prime}$  terminus of the  $5^{\prime}$  UTR as well as the  $5^{\prime}$  and  $3^{\prime}$  termini of the ORF and  $3^{\prime}$  UTR, were revealed to harbor the most informative positions along an mRNA that contribute to half-life (Fig. 5d).

We further interrogated our ISM scores to identify the most pertinent motifs associated with changes in half-life using TF-MoDISco [83]. The poly-A binding element ("AAAAAAA"), ARE ("UAUUUAU"), PUF element ("UGUAHAUA"), putative ELAVL1/2 element ("UUUAU"), polyadenylation element ("AAUAAA"), and m6A motif ("GGACU") were enriched as significant motifs in the  $3^{\prime}$  UTR. Of these, the m6A motif and PUF element were also enriched in the ORF and  $5^{\prime}$  UTR, and the putative ELAVL1/2 and polyadenylation elements were enriched in

the  $5^{\prime}$  UTR (Additional file 1: Fig. S6d). Visualizing ISM scores for two random examples, the  $3^{\prime}$ -UTR segments from TUBGCP3 and PI4K2B, illustrates several learned properties involving these motifs (Fig. 5e). Broadly speaking, a mutation abating a core ARE or PUF element led to a predicted increase in mRNA half-life. Conversely, several point mutations led to a gain of a motif, leading to a predicted decrease in half-life. While a subset of these mutations generated novel motifs, others changed an existing motif closer to its consensus sequence. In some cases, a mutation caused a dual loss of overlapping PUF/ARE motifs, leading to a more severe predicted change in half-life relative to the loss of each individual motif (Fig. 5e).

Next, we evaluated how the model would behave if we inserted either a splice site or one of our five enriched motifs along the full length of each mRNA. We performed this insertional analysis for most human mRNAs in our dataset, and then averaged the result according to the spatial bin of the insertion along each functional region. Consistent with their known roles, we observed that insertion of an m6A site, ARE, or PUF element reduced mRNA stability, with the greatest effect size arising from a  $3^{\prime}$ -UTR insertion (Fig. 5f). In contrast, insertion of a splice site, polyadenylation element, or C-rich element led to enhanced mRNA stability. Consistent with the lasso regression model, the presence of a splice site led to at least a four-fold enhancement in half-life relative to alternative motifs. The most novel property captured in the deep learning model—yet absent from our lasso model—is the strong dependence between the spatial coordinate of the motif along the mRNA and its predicted impact on mRNA half-life, both across and within functional regions (Fig. 5f). For instance, ARE and PUF sites are predicted to most strongly repress mRNA stability if they occur at the  $5^{\prime}$  or  $3^{\prime}$  terminus of the  $3^{\prime}$  UTR, reminiscent of a well-known property of microRNA-mediated repression [24,85]. In contrast, the m6A

element is predicted to most destabilize mRNA immediately after the stop codon, mirroring the known relationship between m6A deposition and mRNA stability [58].

Given the mechanistic link between codon usage and mRNA half-life [8,26-30], we sought to ascertain whether Saluki has also learned this property. We therefore reiterated our insertional analysis, this time using 61 codons (excluding the 3 stop codons) inserted into the first reading frame along the length of each ORF. As before, the model attributed substantially different effect sizes to codons depending upon their position along an ORF, with the greatest predicted effects occurring close to the start and stop codons (Fig. 5g). Although most codons had the greatest predicted effect when inserted closer to the stop codon, several codons such as "CUG", "UUC", "GAA", and "GAC" were predicted to have a modest effect when inserted into most regions except near the start codon (Fig. 5g). We sought to compare our predicted codon effects to existing measures of codon influence, such as the aforementioned CSCs [26-30]. We therefore investigated the relationship between the mean codon influence across an ORF, as predicted by Saluki, relative to the mean CSC across numerous cell types as quantified previously [26]. Reassuringly, there was a strong relationship between the two metrics (Pearson correlation = 0.68, Fig. 5g), suggesting that Saluki successfully captures the influence of codon usage.

# Prediction of  $3^{\prime}$ -UTR regulatory function and genetic variant effects

Given Saluki's strong performance in predicting endogenous half-lives, we sought to evaluate its ability to predict the effect of mRNA sequence and genetic variants therein on mRNA stability in a more controlled context. Massively parallel reporter assays (MPRAs) provide an ideal setting to test causal relationships, because they can directly assess how specific sequences influence

reporter expression. Several studies have deployed MPRAs to test the functional impact of thousands of  $3^{\prime}$ -UTR fragments and genetic variants therein on mRNA stability [86-88]. We performed in silico versions of these MPRA experiments to evaluate their consistency with in vivo experiments.

The first study performed three types of MPRA experiments: i) evaluating the impact of mutation on 8-nt intervals tiling the CXCL2  $3^{\prime}$  UTR, ii) performing a saturation mutagenesis of a specific region within the same  $3^{\prime}$  UTR to measure variant effects, and iii) testing the effect of 3,000 highly conserved  $3^{\prime}$ -UTR segments on RNA stability [86]. To compare our predictions to the first of the three MPRA experiments, we compared our pre-computed ISM scores for the CXCL2  $3^{\prime}$  UTR to those observed in the experiment (Fig. 6a). We observed a general qualitative agreement between model predictions and experiment in the  $3^{\prime}$ -UTR regions showing high activity, which typically overlapped conserved and ARE-containing regions (Fig. 6a). However, there were several novel elements detected by the assay which were conserved but not predicted by the model, indicating several false negatives among the predictions. Next, we evaluated how well our predictions agree with saturation mutagenesis data from the tested CXCL2  $3^{\prime}$ -UTR region. To more closely simulate this experiment in silico, we integrated each of the measured oligonucleotide fragments into the  $3^{\prime}$  UTR of eGFP within the corresponding BTV vector tested in the experiment [86]. Then, for each mutation, we calculated the predicted variant effect as the divergence of predicted half-lives between the reference and alternative sequence. We observed a strong agreement between the observed and predicted variant effects, in which both methods highlighted a strong effect for a set of upstream AREs and a weaker effect for a downstream ARE (Fig. 6b). Collectively, the variant effect predictions agreed with those observed with a

Spearman correlation of 0.69 (Fig. 6c). Finally, we assessed how well we could predict the observations of an MPRA testing the effect of 3,000 highly conserved  $3^{\prime}$ -UTR segments on RNA stability. We achieved a Spearman correlation of 0.63 between our model predictions and experiment (Fig. 6d), suggesting that our model is able to integrate the causal relationship between RNA sequence and RNA stability.


a


![image](https://cdn-mineru.openxlab.org.cn/result/2026-01-06/a0718be8-98e0-4753-8746-6ff7dd4e8c2f/3efc693239c2a3d33c6cab9d9d33b20db01a49a31dcfe45c11e04910bd143460.jpg)



b


![image](https://cdn-mineru.openxlab.org.cn/result/2026-01-06/a0718be8-98e0-4753-8746-6ff7dd4e8c2f/c17adb54a2afb48cc76c904e092f36ef883df88c7e9356cadffc8b175b40b941.jpg)



C


![image](https://cdn-mineru.openxlab.org.cn/result/2026-01-06/a0718be8-98e0-4753-8746-6ff7dd4e8c2f/f601ac0bbc9d4cb4c3122e7fff9036fea91b442a9ffed782ae1ed13406527d0e.jpg)



d


![image](https://cdn-mineru.openxlab.org.cn/result/2026-01-06/a0718be8-98e0-4753-8746-6ff7dd4e8c2f/aa55e87b5cea77b16ef5022170f96f5d5723874655b331fc11fe9131b7457ce6.jpg)



e


![image](https://cdn-mineru.openxlab.org.cn/result/2026-01-06/a0718be8-98e0-4753-8746-6ff7dd4e8c2f/3a7711f27263d5cbf73a9f822d0e8af19801e61f538a17345fba6ce35dbeccca.jpg)



Figure 6. Concordance of Saluki predictions and functional data from massively parallel reporter assays.


a) Effect of mutation on RNA stability, as measured by an MPRA [86], for tiles along the CXCL2  $3^{\prime}$  UTR separated by 8-nt intervals. Also shown are variant effect predictions from Saluki (smoothed along a local 8-nt window) for the same region, and vertebrate base conservation as measured by PhyloP [89]. Predicted AREs are boxed in red, and novel elements detected by the MPRA are boxed in orange. b) Saturation mutagenesis of a segment of the CXCL2  $3^{\prime}$  UTR, boxed in purple in part (a). Shown are the observed variant effects (top) and Saluki's predicted variant effects (bottom). The reference sequence is shown for each, in which the nucleotide height is scaled according to the mean observed or predicted effect for that position. c) Scatter plot of the observed and predicted variant effects shown in panel (b). d) Scatter plot of the observed and predicted  $3^{\prime}$ -UTR effects for each of 3,000 conserved  $3^{\prime}$  UTRs profiled by fastUTR [86]. e) Scatter plot of the observed and predicted variant effects, as measured in Beas2B cells [87]. Also indicated are the Pearson (r) and Spearman (rho) correlation values for panels (c-e).

To further evaluate the generality of our model, we turned to other MPRAs testing both  $3^{\prime}$ -UTR and variant effects in Jurkat and Beas2B cells; the elements tested were heavily enriched in ARE-containing regions [87]. We observed Spearman correlations of 0.42 and 0.49 between our model predictions and observed  $3^{\prime}$ -UTR effects in Jurkat and Beas2B cells, respectively (Additional file 1: Fig. S7a,b). These predictions were similar to the observed  $3^{\prime}$ -UTR effects between both cell types (Spearman correlation = 0.52, Additional file 1: Fig. S7c), suggesting that the model was about as predictive as the same MPRA experiment performed in different cell types. Similarly, when evaluating variant effects in both cell types, we achieved Spearman correlations of 0.31 and 0.36 between our model predictions and observed variant effects in Jurkat and Beas2B cells, respectively (Fig. 6e and Additional file 1: Fig. S7d,e). Yet again, this predictive performance was on par with the agreement of observed variant effects between the two cell types (Spearman correlation = 0.26, Additional file 1: Fig. S7f).

As a final test for the model, we evaluated its agreement with MPRAs that measured the functional effect of 12,173  $3^{\prime}$  UTRs in six cell types [88]. While the observed values agreed well between any pair of cell types (Spearman correlations from 0.60 to 0.83), our predictions agreed a bit more modestly (Spearman correlations from 0.26 to 0.50, Additional file 1: Fig. S7f), suggesting that they partially captured the causal factors linking RNA sequence to stability in this dataset.

# DISCUSSION

Since the emergence of the first transcriptome-wide measurements, there have been numerous efforts to understand the sequence-encoded determinants of mRNA half-life [6-8,38-40,90,91].

Despite the diversity of modeling approaches developed to predict half-life, the quality of the predictive engines has been severely limited by the quality of the measurement itself. It has long been recognized that different methodologies suffer from biased measurement [16,17], leading to an inconsistent portrait of transcriptome-wide half-lives. Indeed, nearly three decades ago, it was theorized that considering an ensemble of different methods would ultimately lead to a more precise definition of half-life [18]. In this study, we aimed to establish a more precise “ground truth” of mammalian mRNA half-lives, empowered by the systematic collection and subsequent meta-analysis of a large cohort of mammalian half-life datasets. As anticipated, this exercise revealed inconsistencies among data derived from different research groups and experimental protocols (e.g., pulse-chase vs transcriptional shutoff). We therefore derived consensus measurements of mRNA half-life which were less susceptible to technical noise and methodological bias.

Importantly, these bias-adjusted measurements led to several favorable outcomes. First, there was an improved interspecies agreement between the half-lives of orthologous human and mouse genes, indicating a greater conservation of this molecular trait than previously appreciated. Next, there was a substantial improvement in quantitative efforts to predict half-life from genetic and biochemical features. Finally, the relative importance of the heterogeneous mechanisms influencing half-life could be better delineated. Our lasso regression and deep learning models captured most of the known determinants of half-life, integrating the influence of AREs, PUF elements, YTHDF1-binding elements, splice sites, microRNA binding sites, and codon composition. While our lasso models implicate novel roles for several additional RBPs, our deep

learning models propose novel properties of gene regulation such as the position-dependent effects of codons, motifs, and splice sites.

Aside from its improved accuracy, one of the key advantages of our deep learning model relative to our lasso regression models is its ability to rapidly generate predictions given arbitrary sequences, obviating the need to compute hundreds to thousands of features for each mRNA. This property makes Saluki amenable to the rapid prediction of variant effects, enabling us to produce a global map of predicted changes in half-life for all possible mutations in the exons of protein-coding genes. Given the concordance between our in silico predictions as well as in vivo MPRA measurements of regulatory element and variant function, our work offers a promising initial foray into the problem of predicting the post-transcriptional consequences of genetic mutation in exonic sequences.

# CONCLUSIONS

In this study, we demonstrated our ability to predict mRNA half-life from sequence with an  $r^2$  of 0.59, which represents nearly a  $50\%$  performance improvement relative to existing models in mammals ( $r^2 = 0.20$  and  $r^2 = 0.39$ ) [6,7]. Among the biggest reasons for this improvement is simply the derivation of a more precise measurement of mRNA half-life from a meta-analysis of transcriptome-wide half-life datasets. Given our encouraging results in the auxiliary tasks of sequence and variant effect prediction, as quantified by the consistency between model predictions and MPRA data, we foresee several practical applications for Saluki.

First, there is a critical need to bioengineer stable RNAs for vaccine development and gene therapy applications. Despite the recent success of mRNA vaccines targeting Covid-19, they are inherently unstable and prone to degradation due to the prevalence of intracellular and extracellular RNases [92]. Problems surrounding mRNA instability and inefficient protein expression have been partially overcome through the incorporation of stabilizing mRNA structures [15], a  $5^{\prime}$ -cap, modified nucleosides, and codon optimality rules [92]. In the context of adeno-associated virus (AAV)-based gene therapy vectors, codon optimality rules have also been engineered to encourage greater mRNA stabilities and higher translation rates for several therapeutic proteins [93]. Nevertheless, the global sequence has not yet been jointly optimized for the mRNA to exhibit a longer half-life. Deep learning models have been successfully coupled to generative models, such as generative adversarial networks, to engineer nucleic acids to obey certain properties. This is exemplified by the design of  $5^{\prime}$  UTRs to improve translation rate [94],  $3^{\prime}$  UTRs to enhance cleavage and polyadenylation [95,96], and coding sequences to evolve

antimicrobial peptides [97]. Similarly, we envision Saluki's use as an external function analyzer, or "oracle", to engineer mRNAs with desired half-life properties.

Given the widespread roles of mRNA decay in the genetics of human health and disease [12,88,98], Saluki is also poised to provide insight into whether a genetic variant functions through post-transcriptional gene regulatory mechanisms. Towards this goal, it could help to fine-map causal eQTLs through the integration of its score into supervised learning frameworks [13]. Such scores could also be integrated into tools intended to identify pathogenic non-coding variants in the human genome [14] to further unravel the genetic basis of disease.

# METHODS

mRNA half-life data collection and pre-processing

We manually collected the processed half-life values from all of the studies indicated (Table 1). In cases in which genes were provided as gene names or RefSeq IDs, the names were converted to their corresponding Ensembl IDs using information from the Ensembl BioMart. Half-life measurements were log-transformed after adding a pseudocount of 0.1 or 1, depending upon whether the unit of the provided half-life was measured in hours or minutes, respectively. Duplicated IDs were then averaged to compute a single half-life measurement for each gene. Five human datasets from four studies provided data as degradation rates rather than half-lives [29,38,46,49], so their transformed half-life values were negated. Finally, we generated a sparse matrix of half-lives for all genes x samples, containing missing values for genes in which the sample did not provide a measured half-life (Additional file 2: Dataset S1).

To evaluate the relatedness between samples, we first extracted the subset of genes in the aforementioned matrix for which  $\geq 10$  human samples (or  $\geq 5$  mouse samples) reported non-missing values. We then z-score transformed the half-lives from each sample to standardize the scale of each sample. To impute the missing values of the matrix, we used the estim_ncpPCA function in the missMDA R package to estimate the appropriate number of principal components (PC) for the imputePCA function, considering between 0 to 20 components for the human samples (or 0 to 10 components for the mouse samples). Finally, the data was quantile normalized using the normalizequantiles function in the preprocessCore R package. The first PC of this imputed matrix (as computed by the prcomp function in R) was used as a robust cell-type-independent measurement of mRNA half-life, and computed for human and mouse

species separately. For human, the samples from a single study [44] were removed prior to computing the PC because the samples from this study represented large outliers relative to all other samples (Additional file 3: Dataset S2).

For evolutionary comparisons between species, one-to-one human-to-mouse orthologs were acquired from the Ensembl v90 BioMart [99] by extracting the "Mouse gene stable ID" and "Mouse homology type" with respect to each human gene.

To examine sample relatedness, the first two PCs were computed using the transpose of the imputed (gene x sample) matrix, and samples were annotated and colored by their cell type of origin, study of origin, and half-life measurement technique.

# Gene annotation set

Gene annotations for protein coding genes were derived from Ensembl v83 (hg38 genome build) and v90 (mm10 genome build) for human and mouse, respectively [99]. Only protein-coding genes were carried forward for analysis. Out of all transcripts corresponding to each gene, the one with the longest ORF, followed by the longest  $5^{\prime}$  UTR, followed by the longest  $3^{\prime}$  UTR was chosen as the representative transcript for that gene [2,24,25].

# Basic mRNA features to predict half-life

The G/C content and lengths of each of these functional regions (i.e.,  $5^{\prime}$  UTRs, ORFs, and  $3^{\prime}$  UTRs), intron length, and ORF exon junction density (computed as the number of exon junctions per kilobase of ORF sequence) were gathered as "basic" mRNA features associated with mRNA

half-life [2,6,7]. All length-related features were transformed such that:  $\hat{x} \gets \log_{10}(x + 0.1)$  to reduce the right skew [2].

# Sequence features to predict mRNA half-life

All genetically encoded features are summarized in Table 2. Codon frequencies were extracted from ORF sequences using the oligonucleotideFrequency function (parameters width=3, step=3) in the Biostrings R package, and normalizing the counts for each codon by the sum of all codon counts for the gene. K-mer frequencies were computed similarly from each of the  $5^{\prime}$  UTR, ORF, and  $3^{\prime}$  UTR sequences (parameters width={1..7} depending on the size of the k-mer, step=1).

MicroRNA features were collected for mammalian-conserved miRNA families using TargetScanHuman7.2 and TargetScanMouse7.2 [24]. We computed a miRNA target binding score by negating the “cumulative weighted context++ score” for each miRNA family.

The degree of predicted binding to a number of RBPs was computed for  $5^{\prime}$  UTR, ORF, and  $3^{\prime}$  UTR sequences separately using SeqWeaver [68] and DeepRiPE [69]. For SeqWeaver, we generated predictions on each 50 nt window, padding the sequence with its neighboring 475 nt upstream and downstream sequence (or Ns in the case of sequences at the boundary of a region) to generate a 1000 nt input sequence. For DeepRiPE, we generated predictions on each 50 nt window, padding the sequence with its neighboring 50 nt upstream and downstream sequence (or Ns in the case of sequences at the boundary of a region) to generate a 150 nt input sequence. DeepRiPE additionally required information about the functional region of interest, which we provided according to the functional region being considered. After generating predictions for all

human and mouse RBPs for each functional region and gene, we computed the average value of the binding by normalizing the sum of values for each predicted RBP to the total length of the functional region.

# Biochemical features

All biochemical features are summarized in Table 2. The number of CLIP peaks associated with each gene for each of 133 RBPs was downloaded from the ENCORI database [71]. eCLIP peaks were collected from the ENCORE browser as narrowPeak BED files [72]. The peaks for each RBP were intersected with gene body annotations and counted for each gene using “bedtools intersect” (parameters -s -c) [100]. PAR-CLIP peaks were downloaded from a previous study [70] and processed similarly to count the total peaks overlapping the gene body. m6A pathway (i.e., a6A, m6Am, YTHDF2, METTL3, METTL14, and WTAP) CLIP peaks were collected from an assortment of previous studies [37,42,73–77], and if not already provided, the number of peaks intersecting gene bodies was computed. For all CLIP assays, genes with missing values (i.e., those without an annotated peak) were considered to have zero peaks. All peak counts were transformed as such:  $\hat{x} \gets \log_{10}(x + 1)$  to reduce the right skew of the distributions.

Processed RIP-seq and translational efficiency data was downloaded from previous studies [36,78-80]. Merging these values with half-lives resulted in missing values; these were imputed using the impute function of the imputeR R package (parameter lmFun="plsR") when considered alongside the "basic" continuous features describing mRNA properties.

# Lasso regression modeling

All features being considered in a model (i.e., basic, genetic, biochemical, and/or a subset within each category) and half-life values were concatenated together and z-score normalized by subtracting their respective mean values and dividing by their standard deviations. We trained a lasso regression model on each of 10 folds of the data. A lasso regression model was chosen specifically because it employs an L1 regularization penalty, which leads to the selection of the fewest features that maximally explain the data. The strength of the regularization was controlled by a single  $\lambda$  parameter, which was optimized using 10-fold CV on the entire dataset. To evaluate the usefulness of different subsets of features in improving predictive performance, we evaluated the Pearson correlation of the predictions of the lasso regression model on each of the 10 held-out folds of data, and performed paired t-tests to evaluate significant improvements in performance. To interpret the best model, we trained the lasso regression model on the full dataset and visualized the top 30 coefficients with the greatest magnitude.

# Saluki model architecture and training

We trained a hybrid convolutional and recurrent deep neural network to predict half-life from its spliced mRNA sequence with several important gene structure annotations. Following previous work on applying such models to DNA/RNA sequence, we one-hot encoded the nucleotide sequence to four input tracks. Due to the strong influence of splicing on RNA stability, we added a fifth binary track to mark the positions of exon junctions at each exon's  $5^{\prime}$  nucleotide. Due to the strong influence of mRNA codon composition, we added a sixth binary track to mark the beginning nucleotide of each codon, which implicitly labels the  $5^{\prime}$  and  $3^{\prime}$  UTRs due to their absence of codon markers. Although a nucleotide-only strategy may be preferable for mutation

effect prediction, these mRNA features are important and could not easily be predicted by the model without adding substantial auxiliary training information.

Because mRNA lengths vary by orders of magnitude, we designed a model architecture to work with variable length sequences. We capped the length of an input mRNA to the model at 12,288 nt for practical purposes, finding that consideration of longer sequences led to equal performance at the cost of slower training speed. To accommodate the rare scenario in which an mRNA exceeded 12,288 nt, we truncated such cases from the  $5^{\prime}$  end, restricting the model to access the  $3^{\prime}$ -most 12,288 nt. Conversely, for the common scenario in which an mRNA was shorter, we padded such cases with zeros at their  $3^{\prime}$  ends, representing “N’s. We made use of the ten folds of human genes described above, and divided the mouse genes into ten folds that maintain the closest homologous genes in the same fold across species based on Ensembl annotations [99].

Our model architecture, which we refer to as Saluki, consists of a “tower” of six convolution blocks to reach a resolution for which each position represents 128 nt (Fig. 5a and Additional file 1: Fig. S6a). Each block includes the following operations: (i) layer normalization [101], (ii) ReLU activation, (iii) 1D convolution with kernel width 5, (iv) dropout, and (v) max pooling with width 2. Overall, the model consists of 155,521 learnable parameters. We chose layer normalization over batch normalization because most of the  $3^{\prime}$  positions are zero padded and would confuse the batch statistics. In contrast, layer normalization is computed independently at each position and simply maintains a zero vector for the padded regions.

To make a single numeric prediction for each sequence, we must aggregate information across the variable lengths. To achieve this, we use a common recurrent neural network block called the Gated Recurrent Unit (GRU) [102]. After layer normalization and ReLU activation, the GRU runs backward from the often-padded  $3^{\prime}$  end to the information dense  $5^{\prime}$  end. We take the final GRU hidden representation from the most  $5^{\prime}$  position as a summary of the entire sequence. We apply a subsequent dense block, consisting of batch normalization, ReLU, and a dense layer. Finally, we apply one more block of batch normalization, ReLU, and a final dense transformation to produce the half-life prediction.

We trained with the MSE loss function using the Adam optimizer on batches of 64 examples and learning rate 0.001, beta1 0.9, and beta2 0.98. We clipped gradients to a global norm of 0.5. We used dropout probability 0.3 throughout and added L2 regularization on all convolution, GRU, and dense layer weights with coefficient 0.001. We used skopt to optimize these hyperparameters as well as the number of channels throughout the model. The hyperparameters specified here and 64 channels achieved the greatest validation set accuracy and compose our final model. For model training, we performed early stopping after 25 epochs without improvement and took the final parameters that achieved the greatest Pearson correlation on the validation set.

We trained models using both the human and mouse data using a published approach [5], in which all parameters are shared except for two separate final dense blocks for human and mouse. During training, we iterate between interwoven batches of human and mouse genes.

Our ten folds of genes allowed us to train multiple models, in which each fold was held out as a test set (and another was held out as a validation set). Because we observed variance from training run to training run, we trained five replicate models for each held out test fold, producing a total of fifty trained parameter settings. After preliminary analyses to examine the predictions' variance, we averaged the predictions of the five replicates per test set as an ensemble, which improves accuracy and robustness. For all downstream analyses involving Saluki predictions on third-party test sets, we averaged the predictions from all fifty models.

# In silico mutagenesis with Saluki

We performed in silico saturation mutagenesis (ISM) to predict the effect on mRNA half-life of all transcriptomic nucleotides. For each position, we ran three Saluki forward passes, mutating the reference nucleotide to each of the three possible alternative alleles. For each mutation, we compared the half-life prediction to the reference.

For mutations that modify stop codons, we optionally set the coding track  $3^{\prime}$  onwards to zeros. This mode creates disproportionately large effect predictions for these mutations, which can be inconvenient for analyses focused on alternative aspects. We therefore decided not to modify the coding track downstream of the stop codon.

# Insertional motif analysis with Saluki

Using our ISM scores as input, we ran TF-MoDISco [83] on each functional region (i.e.,  $5^{\prime}$  UTR, ORF, and  $3^{\prime}$  UTR) independently to identify the enriched motifs associated with changes in half-life. We selected several of the resulting PWMs to perform an insertional analysis, choosing

the consensus sequence as a representative kmer to insert. We inserted each kmer into one of 50 evenly-divided positional bins along each functional region of a valid mRNA, replacing the reference sequence with the inserted kmer to preserve the length of the mRNA. A valid mRNA was defined as one whose  $5^{\prime}$  UTR length was  $\geq 100\mathrm{nt}$ , ORF length was  $\geq 500\mathrm{nt}$ , and  $3^{\prime}$  UTR length was  $\geq 500\mathrm{nt}$ . For each insertion, we recorded the predicted change in half-life relative to the corresponding wild-type mRNA. Finally, for each of the 150 positional bins, we averaged the predicted changes in half-lives across all valid mRNAs to calculate the average influence of the motif across heterogeneous sequence contexts. We performed insertional analysis identically with the 61 non-stop codons, except that each codon was inserted into the first reading frame within the ORF in which it was inserted.

# DECLARATIONS

# AVAILABILITY OF DATA AND MATERIALS

The code to reproduce the core results of this work is provided under the Apache2.0 open access license at the following links: https://github.com/vagarwal87/saluki_paper (to reproduce figures) and https://github.com/calico/basenji/tree/master/manuscripts/saluki (to train deep learning models).

# COMPETING INTERESTS STATEMENT

V.A. and D.R.K. are employees of Calico Life Sciences.

# FUNDING

This work was funded by Calico Life Sciences LLC.

# AUTHOR CONTRIBUTIONS

V.A. and D.R.K. conceived of the study. V.A. designed and performed most analyses. D.R.K. trained deep learning models. V.A. generated figures and wrote the paper with feedback from D.R.K.

# ACKNOWLEDGMENTS

We thank Han Yuan for help with TF-MoDISco analysis. We also thank Han Yuan, Divyanshi Srivastava, Nimrod Rubenstein, Charles Ledogar, and David Botstein for critical feedback.

# SUPPLEMENTARY INFORMATION

Additional file 1: Figures S1-S7. Supplementary figures S1 to S7.

Additional file 2: Dataset S1. Sparse matrices of human and mouse half-lives collected for each sample, provided along with the corresponding Ensembl gene IDs and gene names. Missing values exist for mRNAs whose half-lives were not measured or provided.

Additional file 3: Dataset S2. Matrices of filtered, transformed, imputed, and quantile-normalized human and mouse half-lives collected for each sample, provided along with the corresponding Ensembl gene IDs and gene names. Also provided are the aggregated half-lives for each species, computed as PC1 of each matrix.


SUPPLEMENTARY FIGURES


![image](https://cdn-mineru.openxlab.org.cn/result/2026-01-06/a0718be8-98e0-4753-8746-6ff7dd4e8c2f/5190fed02da9c008a5bc01b033c2b23785a4007b01813c70658fe48cedea0698.jpg)



Figure S1. Variability in genes reported in different human half-life datasets.



Barplot of the number of genes whose half-lives were reported in each of 54 human samples. Each sample ID is delimited by underscores and listed according to its study of origin (Table 1), measurement method, cell type, and replicate number.



a


![image](https://cdn-mineru.openxlab.org.cn/result/2026-01-06/a0718be8-98e0-4753-8746-6ff7dd4e8c2f/471e3bc5e3e992c6afbd6dfdd638a9bfa15ba59718c358a9749cc7c662bfe947.jpg)



b


![image](https://cdn-mineru.openxlab.org.cn/result/2026-01-06/a0718be8-98e0-4753-8746-6ff7dd4e8c2f/d6e6ca83a0dc73a4ded933a3f6ca7d1921e4a1ae01a9057c1156ae203987dd68.jpg)



Figure S2. Comparison of half-lives in a compendium of mouse datasets.


a) Barplot of the number of genes whose half-lives were reported in each of 27 mouse samples. Each sample ID is delimited by underscores and listed according to its study of origin (Table 1), measurement method, cell type, and replicate number. b). Heatmap of the absolute value of the Spearman correlations measured between half-lives derived from each pair of 27 mouse samples. Samples are clustered using hierarchical clustering according to the indicated dendrogram. Rows are labeled by the study of origin (Table 1) and colored by the cell type of origin and measurement approach.

![image](https://cdn-mineru.openxlab.org.cn/result/2026-01-06/a0718be8-98e0-4753-8746-6ff7dd4e8c2f/da752a25b2ab692a790aa7f5a69bb25b97195e51fc064ca502fa673445d74eea.jpg)



Figure S3. Relationship among human samples.


This figure is similar to Fig. 2a, except that it shows all human samples evaluated in this study.


a


![image](https://cdn-mineru.openxlab.org.cn/result/2026-01-06/a0718be8-98e0-4753-8746-6ff7dd4e8c2f/44b3070763e2e88166e5f719832c7bee95225bae4ed1daa6bff124db456563ee.jpg)



b


![image](https://cdn-mineru.openxlab.org.cn/result/2026-01-06/a0718be8-98e0-4753-8746-6ff7dd4e8c2f/1dd2b1a5aa94654808b43eccb56b384e6270c9e9e5cabd4bc1cbb509ebd5ce12.jpg)



C


![image](https://cdn-mineru.openxlab.org.cn/result/2026-01-06/a0718be8-98e0-4753-8746-6ff7dd4e8c2f/dc7fda6e91ee41a7706b60ddaa8172ff80c70ae2977565c8b048d84565c1ff1e.jpg)



d


![image](https://cdn-mineru.openxlab.org.cn/result/2026-01-06/a0718be8-98e0-4753-8746-6ff7dd4e8c2f/ca48f1c18756632f3c857dd8c96581594aef48cc4a86dba875ebc591ae332e43.jpg)



e


![image](https://cdn-mineru.openxlab.org.cn/result/2026-01-06/a0718be8-98e0-4753-8746-6ff7dd4e8c2f/31476acad2bd388d112051d14220434c92ac1e928368b0ce4ac60f56057e5457.jpg)



Figure S4. Prediction of mouse half-lives using sequence-encoded features.



a-d) These panels are organized in the same fashion as Fig. 3a-d, except that they evaluate features benchmarked upon mouse data rather than human data. e) Performance of trained human and mouse


models on test sets derived from either the same or different species. Due to the imperfect one-to-one miRNA mappings between the human and mouse, miRNA-related coefficients were excluded from the inter-species predictions. Each statistical comparison shown was evaluated with a paired t-test.


a


![image](https://cdn-mineru.openxlab.org.cn/result/2026-01-06/a0718be8-98e0-4753-8746-6ff7dd4e8c2f/fe4e77157ecce99c37717190dfe4e82b2864c43373c2fa4fb036487e097dc57c.jpg)



B: Basic mRNA features (8) G: Genetic (23,005) b: biochemical (371)



b


![image](https://cdn-mineru.openxlab.org.cn/result/2026-01-06/a0718be8-98e0-4753-8746-6ff7dd4e8c2f/47525d58c9d276dd47a8238ed5bfe243440d446c0d8ae7ee4553058d0b515773.jpg)



C


![image](https://cdn-mineru.openxlab.org.cn/result/2026-01-06/a0718be8-98e0-4753-8746-6ff7dd4e8c2f/1e41da93fa699d7aeb761575c40bb488c6bf5f0755e9a36bb38d5cf3f28449c2.jpg)



d


![image](https://cdn-mineru.openxlab.org.cn/result/2026-01-06/a0718be8-98e0-4753-8746-6ff7dd4e8c2f/a26967e61ef2652aa0499f6058c2a56699a6795d61ed7c24bf2b377f4bfbec8f.jpg)



Figure S5. Prediction of mouse half-lives using biochemical and sequence-encoded features.



This figure is organized in the same fashion as Fig. 3-4, except that it evaluates the subset of optimal genetic features (BC3MS model, Fig. 3), optimal biochemical features (BEeM model, Fig. 4), or a combination of both genetic and biochemical features. The features shown in panels (c-d) are described alongside their corresponding source, such as eCLIP, SeqWeaver, or codon features.


![image](https://cdn-mineru.openxlab.org.cn/result/2026-01-06/a0718be8-98e0-4753-8746-6ff7dd4e8c2f/9c47936a1caff06102624d4607c6406bed8895a81015872944ea7bcbf8f083ac.jpg)



Total parameters: 155,777 Trainable parameters: 155,521 Non-trainable parameters: 256


![image](https://cdn-mineru.openxlab.org.cn/result/2026-01-06/a0718be8-98e0-4753-8746-6ff7dd4e8c2f/dd97c89219ef8b6f527f4c1ac69a56b28e24e638932de444b24820a7850eac54.jpg)


![image](https://cdn-mineru.openxlab.org.cn/result/2026-01-06/a0718be8-98e0-4753-8746-6ff7dd4e8c2f/c5e2a122f0e11cd4e2bd8469bcf984389922b2141260956881e4a9bbdb090475.jpg)


![image](https://cdn-mineru.openxlab.org.cn/result/2026-01-06/a0718be8-98e0-4753-8746-6ff7dd4e8c2f/bb15e81783286c956f7485fad4746f1e24a11ca757cd4dae9bf673dd3351dcec.jpg)



Figure S6. Performance and interpretation of Saluki.


a) Complete architecture of the Saluki model. Indicated for each layer is the layer name, number of parameters in parentheses when applicable, and dimensionality of the input and output matrices. b) Performance of trained Saluki models on each of 10 held-out folds of data. Compared is the relative performance between pairs of models, for both human and mouse species, which iteratively consider additional input tracks. Each model is described by a code indicating the input track considered. A description of the code is provided in the key. An improvement in a more complex model relative to a

simpler model was evaluated with a one-sided, paired t-test, adjusted with a Bonferroni correction to account for the total number of hypothesis tests. c) This panel is organized in the same fashion as Fig. 5c, except that it evaluates the performance on mouse data rather than human data. d) Set of enriched motifs discovered by TF-MoDISco [83] in each of the three functional regions of an mRNA.


a


![image](https://cdn-mineru.openxlab.org.cn/result/2026-01-06/a0718be8-98e0-4753-8746-6ff7dd4e8c2f/fc784f554a99a17fbdb905fd8a8e592dedcf773a1def343d8c711ad3a34664f3.jpg)



b


![image](https://cdn-mineru.openxlab.org.cn/result/2026-01-06/a0718be8-98e0-4753-8746-6ff7dd4e8c2f/c9cccf23b218382bcccc6cc4e7ab943eb900913f64d177a61a0ed5dbffb52f17.jpg)



C


![image](https://cdn-mineru.openxlab.org.cn/result/2026-01-06/a0718be8-98e0-4753-8746-6ff7dd4e8c2f/943ab45041b31ca33a0d8da1cb79e6f2a800e2527fb81c45dbe754791d342186.jpg)



d


![image](https://cdn-mineru.openxlab.org.cn/result/2026-01-06/a0718be8-98e0-4753-8746-6ff7dd4e8c2f/52493dedb5d39e0b24b191ac87376804f688f86154f12070d0a31ad3ab7c8c0e.jpg)



e


![image](https://cdn-mineru.openxlab.org.cn/result/2026-01-06/a0718be8-98e0-4753-8746-6ff7dd4e8c2f/9a876d90d4799a48e998f3c24b629abdc50b62eca3b0358309605510637352ee.jpg)



f


![image](https://cdn-mineru.openxlab.org.cn/result/2026-01-06/a0718be8-98e0-4753-8746-6ff7dd4e8c2f/981cb0e058c484b9c02a833ba05e84225392b0d7579591fe12f213eeda54ad60.jpg)



g


![image](https://cdn-mineru.openxlab.org.cn/result/2026-01-06/a0718be8-98e0-4753-8746-6ff7dd4e8c2f/b6a7db0e6f836efc81acd97cc009e98b2ae08fb2ca66659850fc0d6d695e2058.jpg)


# Figure S7. Concordance of Saluki predictions and additional functional data from massively parallel reporter assays.

a-c) Scatter plot of the observed and predicted  $3^{\prime}$ -UTR effects, as measured in a) Jurkat cells or b) Beas2B cells, alongside c) a plot of observed  $3^{\prime}$ -UTR effects between the pair of cell types [87]. d-f) These panels are organized like panels (a-c), but display variant effects instead of  $3^{\prime}$ -UTR effects [87]. Panel e) is identical to that shown in Fig. 6e and reproduced here for convenience. Also indicated are the Pearson (r) and Spearman (rho) correlation values for panels (a-f). g) Scatter matrix displaying scatter plots corresponding to each of the 21 pairs of possible comparisons (lower diagonal elements) involving Saluki predictions as well as the measured stability effects for  $3^{\prime}$ -UTR fragments measured in each of six cell types [88]. Shown on the diagonal is a histogram of the predicted or observed RNA half-life scores. Also shown are Spearman correlation values among each pair of comparisons, with the size of the text proportional to the magnitude of the correlation coefficient (upper diagonal elements).

# REFERENCES



1. Kelley DR, Reshef YA, Bileschi M, Belanger D, McLean CY, Snoek J. Sequential regulatory activity prediction across chromosomes with convolutional neural networks. Genome Res. 2018;28:739-50.





2\. Agarwal V, Shendure J. Predicting mRNA Abundance Directly from Genomic Sequence Using Deep Convolutional Neural Networks. Cell Rep. 2020;31:107663.





3. Zhou J, Theesfeld CL, Yao K, Chen KM, Wong AK, Troyanskaya OG. Deep learning sequence-based ab initio prediction of variant effects on expression and disease risk. Nat Genet. 2018;50:1171-9.





4\. Avsec Ž, Agarwal V, Visentin D, Ledsam JR, Grabska-Barwinska A, Taylor KR, et al. Effective gene expression prediction from sequence by integrating long-range interactions. Nat Methods. 2021;18:1196-203.





5. Kelley DR. Cross-species regulatory sequence activity prediction. PLoS Comput Biol. 2020;16:e1008050.





6\. Sharova LV, Sharov AA, Nedorezov T, Piao Y, Shaik N, Ko MSH. Database for mRNA half-life of 19 977 genes obtained by DNA microarray analysis of pluripotent and differentiating mouse embryonic stem cells. DNA Res. 2009;16:45-58.





7. Spies N, Burge CB, Bartel DP. 3' UTR-isoform choice has limited influence on the stability and translational efficiency of most mRNAs in mouse fibroblasts [Internet]. Genome Research. 2013. p. 2078–90. Available from: http://dx.doi.org/10.1101/gr.156919.113





8\. Cheng J, Maier KC, Avsec Z, Rus P, Gagneur J. Cis-regulatory elements explain most of the mRNA stability variation across genes in yeast. RNA. 2017;23:1648-59.





9. Friedman RC, Farh KK, Burge CB, Bartel DP. Most mammalian mRNAs are conserved targets of microRNAs. Genome Res. 2009;19:92-105.





10\. Lindblad-Toh K, Garber M, Zuk O, Lin MF, Parker BJ, Washietl S, et al. A high-resolution map of human evolutionary constraint using 29 mammals. Nature. 2011;478:476-82.





11. Xie X, Lu J, Kulbokas EJ, Golub TR, Mootha V, Lindblad-Toh K, et al. Systematic discovery of regulatory motifs in human promoters and 3' UTRs by comparison of several mammals. Nature. 2005;434:338-45.





12\. Pai AA, Cain CE, Mizrahi-Man O, De Leon S, Lewellen N, Veyrieras J-B, et al. The contribution of RNA decay quantitative trait loci to inter-individual variation in steady-state gene expression levels. PLoS Genet. 2012;8:e1003000.





13. Wang QS, Kelley DR, Ulirsch J, Kanai M, Sadhuka S, Cui R, et al. Leveraging supervised learning for functionally informed fine-mapping of cis-eQTLs identifies an additional 20,913 putative causal eQTLs. Nat Commun. 2021;12:3394.





14\. Rentzsch P, Witten D, Cooper GM, Shendure J, Kircher M. CADD: predicting the deleteriousness of variants throughout the human genome. Nucleic Acids Res. 2019;47:D886-94.





15. Leppek K, Byeon GW, Kladwang W, Wayment-Steele HK, Kerr CH, Xu AF, et al. Combinatorial optimization of mRNA structure, stability, and translation for RNA-based therapeutics. bioRxiv [Internet].





2021; Available from: http://dx.doi.org/10.1101/2021.03.29.437587





16. Ross J. mRNA stability in mammalian cells. Microbiol Rev. 1995;59:423-50.





17\. Loflin PT, Chen CY, Xu N, Shyu AB. Transcriptional pulsing approaches for analysis of mRNA turnover in mammalian cells. Methods. 1999;17:11-20.





18. Harrold S, Genovese C, Kobrin B, Morrison SL, Milcarek C. A comparison of apparent mRNA half-life using kinetic labeling techniques vs decay following administration of transcriptional inhibitors. Anal Biochem. 1991;198:19-29.





19\. Miller C, Schwalb B, Maier K, Schulz D, Dümcke S, Zacher B, et al. Dynamic transcriptome analysis measures rates of mRNA synthesis and decay in yeast. Mol Syst Biol. 2011;7:458.





20. Miller MR, Robinson KJ, Cleary MD, Doe CQ. TU-tagging: cell type-specific RNA isolation from intact complex tissues. Nat Methods. Nature Publishing Group; 2009;6:439-41.





21\. Sun M, Schwalb B, Schulz D, Pirkl N, Etzold S, Lariviere L, et al. Comparative dynamic transcriptome analysis (cDTA) reveals mutual feedback between mRNA synthesis and degradation [Internet]. Genome Research. 2012. p. 1350-9. Available from: http://dx.doi.org/10.1101/gr.130161.111





22. Courel M, Clément Y, Bossevain C, Foretek D, Vidal Cruchez O, Yi Z, et al. GC content shapes mRNA storage and decay in human cells. Elife [Internet]. 2019;8. Available from: http://dx.doi.org/10.7554/eLife.49708





23\. Bartel DP. Metazoan MicroRNAs. Cell. 2018;173:20-51.





24. Agarwal V, Bell GW, Nam J-W, Bartel DP. Predicting effective microRNA target sites in mammalian mRNAs. Elife [Internet]. 2015;4. Available from: http://dx.doi.org/10.7554/eLife.05005





25\. Agarwal V, Subtelny AO, Thiru P, Ulitsky I, Bartel DP. Predicting microRNA targeting efficacy in Drosophila [Internet]. Genome Biology. 2018. Available from: http://dx.doi.org/10.1186/s13059-018-1504-3





26. Forrest ME, Pinkard O, Martin S, Sweet TJ, Hanson G, Coller J. Codon and amino acid content are associated with mRNA stability in mammalian cells. PLoS One. 2020;15:e0228730.





27\. Narula A, Ellis J, Taliaferro JM, Rissland OS. Coding regions affect mRNA stability in human cells. RNA. 2019;25:1751-64.





28. Presnyak V, Alhusaini N, Chen Y-H, Martin S, Morris N, Kline N, et al. Codon optimality is a major determinant of mRNA stability. Cell. 2015;160:1111-24.





29\. Wu Q, Medina SG, Kushawah G, DeVore ML, Castellano LA, Hand JM, et al. Translation affects mRNA stability in a codon-dependent manner in human cells. Elife [Internet]. 2019;8. Available from: http://dx.doi.org/10.7554/eLife.45396





30. Hia F, Yang SF, Shichino Y, Yoshinaga M, Murakawa Y, Vandenbon A, et al. Codon bias confers stability to human mRNAs. EMBO Rep. EMBO; 2019;20:e48220.





31\. Mauger DM, Cabral BJ, Presnyak V, Su SV, Reid DW, Goodman B, et al. mRNA structure regulates protein expression through changes in functional half-life. Proc Natl Acad Sci U S A. 2019;116:24075-83.





32. Wu X, Bartel DP. Widespread Influence of 3' -End Structures on Mammalian mRNA Processing and Stability. Cell. 2017;169:905-17.e11.





33\. Van Etten J, Schagat TL, Hrit J, Weidmann CA, Brumbaugh J, Coon JJ, et al. Human Pumilio proteins recruit multiple deadenylases to efficiently repress messenger RNAs. J Biol Chem. 2012;287:36370-83.





34. Fabian MR, Frank F, Rouya C, Siddiqui N, Lai WS, Karetnikov A, et al. Structural basis for the recruitment of the human CCR4-NOT deadenylase complex by tristetraprolin. Nat Struct Mol Biol. Nature Publishing Group; 2013;20:735-9.





35\. Du H, Zhao Y, He J, Zhang Y, Xi H, Liu M, et al. YTHDF2 destabilizes m6A-containing RNA through direct recruitment of the CCR4-NOT deadenylase complex [Internet]. Nature Communications. 2016. Available from: http://dx.doi.org/10.1038/ncomms12626





36. Wang X, Lu Z, Gomez A, Hon GC, Yue Y, Han D, et al. N6-methyladenosine-dependent regulation of messenger RNA stability. Nature. 2014;505:117-20.





37\. Zaccara S, Jaffrey SR. A Unified Model for the Function of YTHDF Proteins in Regulating m6A-Modified mRNA. Cell. 2020;181:1582-95.e18.





38. Yang E, van Nimwegen E, Zavolan M, Rajewsky N, Schroeder M, Magnasco M, et al. Decay rates of human mRNAs: correlation with functional characteristics and sequence attributes. Genome Res. 2003;13:1863-72.





39\. Chan LY, Mugler CF, Heinrich S, Vallotton P, Weis K. Non-invasive measurement of mRNA decay reveals translation initiation as the major determinant of mRNA stability. Elife [Internet]. 2018;7. Available from: http://dx.doi.org/10.7554/eLife.32536





40. Blumberg A, Zhao Y, Huang Y-F, Dukler N, Rice EJ, Chivu AG, et al. Characterizing RNA stability genome-wide through combined analysis of PRO-seq and RNA-seq data. BMC Biol. 2021;19:30.





41\. Lugowski A, Nicholson B, Rissland OS. DRUID: a pipeline for transcriptome-wide measurements of mRNA stability. RNA. 2018;24:623-32.





42. Mauer J, Luo X, Blanjoie A, Jiao X, Grozhik AV, Patil DP, et al. Reversible methylation of m6Am in the 5' cap controls mRNA stability. Nature. 2017;541:371-5.





43\. Wachutka L, Caizzi L, Gagneur J, Cramer P. Global donor and acceptor splicing site kinetics in human cells [Internet]. eLife. 2019. Available from: http://dx.doi.org/10.7554/elife.45056





44. Duan J, Shi J, Ge X, Dölken L, Moy W, He D, et al. Genome-wide survey of interindividual differences of RNA stability in human lymphoblastoid cell lines. Sci Rep. 2013;3:1318.





45\. Tani H, Mizutani R, Salam KA, Tano K, Ijiri K, Wakamatsu A, et al. Genome-wide determination of RNA stability reveals hundreds of short-lived noncoding transcripts in mammals. Genome Res. 2012;22:947-56.





46. Kawata K, Wakida H, Yamada T, Tanieke K, Han H, Seki M, et al. Metabolic labeling of RNA using multiple ribonucleoside analogs enables the simultaneous evaluation of RNA synthesis and degradation rates. Genome Res. 2020;30:1481-91.





47\. Schwalb B, Michel M, Zacher B, Frühauf K, Demel C, Tresch A, et al. TT-seq maps the human transient transcriptome. Science. 2016;352:1225-8.





48. Schueler M, Munschauer M, Gregersen LH, Finzel A, Loewer A, Chen W, et al. Differential protein occupancy profiling of the mRNA transcriptome. Genome Biol. 2014;15:R15.





49\. Larsson E, Sander C, Marks D. mRNA turnover rate limits siRNA and microRNA efficacy. Mol Syst Biol. 2010;6:433.





50. Rahmanian S, Balderrama-Gutierrez G, Wyman D, McGill CJ, Nguyen K, Spitale R, et al. Long-TUC-seq is a robust method for quantification of metabolically labeled full-length isoforms [Internet]. bioRxiv. 2020 [cited 2021 Sep 8]. p. 2020.05.01.073296. Available from: https://www.biorxiv.org/content/10.1101/2020.05.01.073296v1.abstract





51\. Arango D, Sturgill D, Alhusaini N, Dillman AA, Sweet TJ, Hanson G, et al. Acetylation of Cytidine in mRNA Promotes Translation Efficiency. Cell. 2018;175:1872-86.e24.





52. Melé M, Mattioli K, Mallard W, Shechner DM, Gerhardinger C, Rinn JL. Chromatin environment, transcriptional regulation, and splicing distinguish lincRNAs and mRNAs. Genome Res. 2017;27:27-37.





53\. Cao J, Zhou W, Steemers F, Trapnell C, Shendure J. Sci-fate characterizes the dynamics of gene expression in single cells. Nat Biotechnol. 2020;38:980-8.





54. Schofield JA, Duffy EE, Kiefer L, Sullivan MC, Simon MD. TimeLapse-seq: adding a temporal dimension to RNA sequencing through nucleoside recoding. Nat Methods. 2018;15:221-5.





55\. Friedel CC, Dölken L, Ruzsics Z, Koszinowski UH, Zimmer R. Conserved principles of mammalian transcriptional regulation revealed by RNA half-life. Nucleic Acids Res. 2009;37:e115.





56. Herzog VA, Reichholf B, Neumann T, Rescheneder P, Bhat P, Burkard TR, et al. Thiol-linked alkylation of RNA to assess expression dynamics. Nat Methods. 2017;14:1198-204.





57\. Eisen TJ, Eichhorn SW, Subtelny AO, Lin KS, McGeary SE, Gupta S, et al. The Dynamics of Cytoplasmic mRNA Metabolism. Mol Cell. 2020;77:786-99.e10.





58. Ke S, Pandya-Jones A, Saito Y, Fak JJ, Vågbø CB, Geula S, et al. m6A mRNA modifications are deposited in nascent pre-mRNA and are not required for splicing but do specify cytoplasmic turnover. Genes Dev. 2017;31:990-1006.





59\. Erhard F, Baptista MAP, Krammer T, Hennig T, Lange M, Arampatzi P, et al. scSLAM-seq reveals core features of transcription dynamics in single cells. Nature. 2019;571:419-23.





60. Geula S, Moshitch-Moshkovitz S, Dominissini D, Mansour AA, Kol N, Salmon-Divon M, et al. Stem cells. m6A mRNA methylation facilitates resolution of naive pluripotency toward differentiation. Science. 2015;347:1002-6.





61\. Zheng X, Yang P, Lackford B, Bennett BD, Wang L, Li H, et al. CNOT3-Dependent mRNA Deadenylation Safeguards the Pluripotent State. Stem Cell Reports. 2016;7:897-910.





62. Dölken L, Ruzsics Z, Rädle B, Friedel CC, Zimmer R, Mages J, et al. High-resolution gene expression profiling for simultaneous kinetic parameter analysis of RNA synthesis and decay. RNA. 2008;14:1959-72.





63\. Rabani M, Levin JZ, Fan L, Adiconis X, Raychowdhury R, Garber M, et al. Metabolic labeling of RNA uncovers principles of RNA production and degradation dynamics in mammalian cells. Nat Biotechnol. 2011;29:436-42.





64. Rabani M, Raychowdhury R, Jovanovic M, Rooney M, Stumpo DJ, Pauli A, et al. High-resolution sequencing and modeling identifies distinct dynamic RNA regulatory strategies. Cell. 2014;159:1698-710.





65\. Schwanhausser B, Busse D, Li N, Dittmar G, Schuchhardt J, Wolf J, et al. Global quantification of mammalian gene expression control. Nature. 2011;473:337-42.





66. Lee JE, Lee JY, Wilusz J, Tian B, Wilusz CJ. Systematic analysis of cis-elements in unstable mRNAs demonstrates that CUGBP1 is a key regulator of mRNA decay in muscle cells. PLoS One. 2010;5:e11201.





67\. Nam J-W, Rissland OS, Koppstein D, Abreu-Goodger C, Jan CH, Agarwal V, et al. Global analyses of the effect of different cellular contexts on microRNA targeting. Mol Cell. 2014;53:1031-43.





68. Park CY, Zhou J, Wong AK, Chen KM, Theesfeld CL, Darnell RB, et al. Genome-wide landscape of RNA-binding protein target site dysregulation reveals a major impact on psychiatric disorder risk. Nat Genet. 2021;53:166-73.





69\. Ghanbari M, Ohler U. Deep neural networks for interpreting RNA-binding protein target preferences. Genome Res. 2020;30:214-26.





70. Mukherjee N, Wessels H-H, Lebedeva S, Sajek M, Ghanbari M, Garzia A, et al. Deciphering human ribonucleoprotein regulatory networks. Nucleic Acids Res. 2019;47:570-81.





71\. Li J-H, Liu S, Zhou H, Qu L-H, Yang J-H. starBase v2.0: decoding miRNA-ceRNA, miRNA-ncRNA and protein-RNA interaction networks from large-scale CLIP-Seq data. Nucleic Acids Res. Oxford Academic; 2013;42:D92-7.





72. Van Nostrand EL, Freese P, Pratt GA, Wang X, Wei X, Xiao R, et al. A large-scale binding and functional map of human RNA-binding proteins. Nature. 2020;583:711-9.





73\. Park OH, Ha H, Lee Y, Boo SH, Kwon DH, Song HK, et al. Endoribonucleolytic Cleavage of m6A-Containing RNAs by RNase P/MRP Complex. Mol Cell. 2019;74:494-507.e8.





74. Liu J, Yue Y, Han D, Wang X, Fu Y, Zhang L, et al. A METTL3-METTL14 complex mediates mammalian nuclear RNA N6-adenosine methylation. Nat Chem Biol. Springer Science and Business Media LLC; 2014;10:93-5.





75\. Shu X, Cao J, Cheng M, Xiang S, Gao M, Li T, et al. A metabolic labeling method detects m6A transcriptome-wide at single base resolution. Nat Chem Biol. Springer Science and Business Media LLC; 2020;16:887-95.





76. Schwartz S, Mumbach MR, Jovanovic M, Wang T, Maciag K, Bushkin GG, et al. Perturbation of m6A writers reveals two distinct classes of mRNA methylation at internal and 5' sites. Cell Rep. 2014;8:284-96.





77\. Batista PJ, Molinie B, Wang J, Qu K, Zhang J, Li L, et al. m6A RNA Modification Controls Cell Fate Transition in Mammalian Embryonic Stem Cells. Cell Stem Cell. 2014;15:707-19.





78. Hendrickson DG, Kelley DR, Tenen D, Bernstein B, Rinn JL. Widespread RNA binding by chromatin-associated proteins. Genome Biol. 2016;17:28.





79\. Rissland OS, Subtelny AO, Wang M, Lugowski A, Nicholson B, Laver JD, et al. The influence of





microRNAs and poly(A) tail length on endogenous mRNA-protein complexes [Internet]. Genome Biology. 2017. Available from: http://dx.doi.org/10.1186/s13059-017-1330-z





80. Wang X, Zhao BS, Roundtree IA, Lu Z, Han D, Ma H, et al. N(6)-methyladenosine Modulates Messenger RNA Translation Efficiency. Cell. 2015;161:1388-99.





81\. Agarwal V, Lopez-Darwin S, Kelley DR, Shendure J. The landscape of alternative polyadenylation in single cells of the developing mouse embryo. Nat Commun. 2021;12:5101.





82. Tian B, Graber JH. Signals for pre-mRNA cleavage and polyadenylation. Wiley Interdiscip Rev RNA. 2012;3:385-96.





83\. Shrikumar A, Tian K, Avsec Z, Shcherbina A, Banerjee A, Sharmin M, et al. Technical Note on Transcription Factor Motif Discovery from Importance Scores (TF-MoDISco) version 0.5.6.5 [Internet]. arXiv [cs.LG]. 2018. Available from: http://arxiv.org/abs/1811.00416





84. Kelley DR, Snoek J, Rinn J. Basset: Learning the regulatory code of the accessible genome with deep convolutional neural networks. 2015;





85\. Grimson A, Farh KK, Johnston WK, Garrett-Engele P, Lim LP, Bartel DP. MicroRNA targeting specificity in mammals: determinants beyond seed pairing. Mol Cell. 2007;27:91-105.





86. Zhao W, Pollack JL, Blagev DP, Zaitlen N, McManus MT, Erle DJ. Massively parallel functional annotation of 3' untranslated regions. Nat Biotechnol. 2014;32:387-91.





87\. Siegel DA, Le Tonqueze O, Biton A, Zaitlen N, Erle DJ. Massively parallel analysis of human  $3^{\prime}$  UTRs reveals that AU-rich element length and registration predict mRNA destabilization. G3 Genes|Genomes|Genetics [Internet]. Oxford Academic; 2021 [cited 2022 Feb 22];12. Available from: https://academic.oup.com/g3journal/article-abstract/12/1/jkab404/6446033





88. Griesemer D, Xue JR, Reilly SK, Ulirsch JC, Kukreja K, Davis JR, et al. Genome-wide functional screen of  $3^{\prime}$  UTR variants uncovers causal variants for human disease and evolution. Cell. Elsevier; 2021;184:5247-60.





89\. Navarro Gonzalez J, Gonzalez JN, Zweig AS, Speir ML, Schmelter D, Rosenbloom KR, et al. The UCSC Genome Browser database: 2021 update [Internet]. Nucleic Acids Research. 2020. Available from: http://dx.doi.org/10.1093/nar/gkaa1070





90. Lam LT, Pickeral OK, Peng AC, Rosenwald A, Hurt EM, Giltnane JM, et al. Genomic-scale measurement of mRNA turnover and the mechanisms of action of the anti-cancer drug flavopiridol. Genome Biol. 2001;2:RESEARCH0041.





91\. Raghavan A, Ogilvie RL, Reilly C, Abelson ML, Raghavan S, Vasdewani J, et al. Genome-wide analysis of mRNA decay in resting and activated primary human T lymphocytes. Nucleic Acids Res. Oxford Academic; 2002;30:5529-38.





92. Park JW, Lagniton PNP, Liu Y, Xu R-H. mRNA vaccines for COVID-19: what, why and how. Int J Biol Sci. 2021;17:1446-60.





93\. Colella P, Ronzitti G, Mingozzi F. Emerging Issues in AAV-Mediated In Vivo Gene Therapy. Mol Ther Methods Clin Dev. 2018;8:87-104.





94. Sample PJ, Wang B, Reid DW, Presnyak V, McFadyen IJ, Morris DR, et al. Human  $5^{\prime}$  UTR design





and variant effect prediction from a massively parallel translation assay. Nat Biotechnol. Nature Publishing Group; 2019;37:803-9.





95. Linder J, Bogard N, Rosenberg AB, Seelig G. A Generative Neural Network for Maximizing Fitness and Diversity of Synthetic DNA and Protein Sequences. Cell Syst. 2020;11:49-62.e16.





96\. Bogard N, Linder J, Rosenberg AB, Seelig G. A Deep Neural Network for Predicting and Engineering Alternative Polyadenylation. Cell. 2019;178:91-106.e23.





97. Gupta A, Zou J. Feedback GAN for DNA optimizes protein functions. Nature Machine Intelligence. 2019;1:105-11.





98\. Hollams EM, Giles KM, Thomson AM, Leedman PJ. MRNA stability and the control of gene expression: implications for human disease. Neurochem Res. 2002;27:957-80.





99. Aken BL, Achuthan P, Akanni W, Amode MR, Bernsdorff F, Bhai J, et al. Ensembl 2017. Nucleic Acids Res. 2016;45:D635-42.





100\. Quinlan AR, Hall IM. BEDTools: a flexible suite of utilities for comparing genomic features. Bioinformatics. 2010;26:841-2.





101. Ba JL, Kiros JR, Hinton GE. Layer normalization. arXiv preprint arXiv:160706450 [Internet]. arxiv.org; 2016; Available from: http://arxiv.org/abs/1607.06450





102\. Chung J, Gulcehre C, Cho K, Bengio Y. Empirical Evaluation of Gated Recurrent Neural Networks on Sequence Modeling [Internet]. arXiv [cs.NE]. 2014. Available from: http://arxiv.org/abs/1412.3555

